// @angular-mdl/core/esm2020/lib/tabs/mdl-tabs.module.mjs
import { MdlTabsComponent } from "./mdl-tabs.component";
import { MdlTabPanelComponent, MdlTabPanelContentComponent, } from "./mdl-tab-panel.component";
import { MdlTabPanelTitleComponent } from "./mdl-tab-panel-title.component";
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { MdlCommonsModule } from "../common/mdl-common.module";
import { MdlRippleModule } from "../ripple/mdl-ripple.module";
import * as i0 from "@angular/core";
export * from "./mdl-tabs.component";
export * from "./mdl-tab-panel.component";
export * from "./mdl-tab-panel-title.component";
const MDL_TABS_DIRECTIVES = [
    MdlTabsComponent,
    MdlTabPanelComponent,
    MdlTabPanelTitleComponent,
    MdlTabPanelContentComponent,
];
export class MdlTabsModule {
    static forRoot() {
        return {
            ngModule: MdlTabsModule,
            providers: [],
        };
    }
}
MdlTabsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlTabsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlTabsModule, declarations: [MdlTabsComponent,
        MdlTabPanelComponent,
        MdlTabPanelTitleComponent,
        MdlTabPanelContentComponent], imports: [MdlCommonsModule, MdlRippleModule, CommonModule, MdlRippleModule], exports: [MdlTabsComponent,
        MdlTabPanelComponent,
        MdlTabPanelTitleComponent,
        MdlTabPanelContentComponent] });
MdlTabsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabsModule, imports: [MdlCommonsModule, MdlRippleModule, CommonModule, MdlRippleModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [MdlCommonsModule, MdlRippleModule, CommonModule, MdlRippleModule],
                    exports: MDL_TABS_DIRECTIVES,
                    declarations: [...MDL_TABS_DIRECTIVES],
                }]
        }] });
