// @angular-mdl/core/esm2020/lib/tabs/mdl-tab-panel.component.mjs
import { Component, ContentChild, HostBinding, Input, ViewEncapsulation, } from "@angular/core";
import { MdlTabPanelTitleComponent } from "./mdl-tab-panel-title.component";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MdlTabPanelContentComponent {
}
MdlTabPanelContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabPanelContentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlTabPanelContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlTabPanelContentComponent, selector: "mdl-tab-panel-content", ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabPanelContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-tab-panel-content",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }] });
export class MdlTabPanelComponent {
    constructor() {
        this.isTabPanel = true;
        this.isActive = false;
    }
}
MdlTabPanelComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabPanelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlTabPanelComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlTabPanelComponent, selector: "mdl-tab-panel", inputs: { title: ["mdl-tab-panel-title", "title"], disabled: "disabled" }, host: { properties: { "class.mdl-tabs__panel": "this.isTabPanel", "class.is-active": "this.isActive" } }, queries: [{ propertyName: "titleComponent", first: true, predicate: MdlTabPanelTitleComponent, descendants: true, static: true }], ngImport: i0, template: `
    <ng-content
      *ngIf="titleComponent"
      select="mdl-tab-panel-content"
    ></ng-content>
    <ng-content *ngIf="!titleComponent"></ng-content>
  `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabPanelComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-tab-panel",
                    template: `
    <ng-content
      *ngIf="titleComponent"
      select="mdl-tab-panel-content"
    ></ng-content>
    <ng-content *ngIf="!titleComponent"></ng-content>
  `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { titleComponent: [{
                type: ContentChild,
                args: [MdlTabPanelTitleComponent, { static: true }]
            }], title: [{
                type: Input,
                args: ["mdl-tab-panel-title"]
            }], disabled: [{
                type: Input,
                args: ["disabled"]
            }], isTabPanel: [{
                type: HostBinding,
                args: ["class.mdl-tabs__panel"]
            }], isActive: [{
                type: HostBinding,
                args: ["class.is-active"]
            }] } });
