// @angular-mdl/core/esm2020/lib/tabs/mdl-tab-panel-title.component.mjs
import { Component, ViewContainerRef, ViewEncapsulation } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlTabPanelTitleComponent {
    constructor(vcRef) {
        this.vcRef = vcRef;
    }
}
MdlTabPanelTitleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabPanelTitleComponent, deps: [{ token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Component });
MdlTabPanelTitleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlTabPanelTitleComponent, selector: "mdl-tab-panel-title", ngImport: i0, template: ` <ng-content></ng-content> `, isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTabPanelTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-tab-panel-title",
                    template: ` <ng-content></ng-content> `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }]; } });
