// @angular-mdl/core/esm2020/lib/layout/mdl-layout-mediator.service.mjs
import { Injectable } from "@angular/core";
import { Subject } from "rxjs";
import * as i0 from "@angular/core";
export class MdlLayoutMediatorService {
    constructor() {
        this.tabMouseoverSubject = new Subject();
        this.tabMouseoutSubject = new Subject();
        this.tabSelectedSubject = new Subject();
    }
    onTabMouseover() {
        return this.tabMouseoverSubject.asObservable();
    }
    tabMouseover(tab) {
        this.tabMouseoverSubject.next(tab);
    }
    onTabMouseOut() {
        return this.tabMouseoutSubject.asObservable();
    }
    tabMouseout(tab) {
        this.tabMouseoutSubject.next(tab);
    }
    onTabSelected() {
        return this.tabSelectedSubject.asObservable();
    }
    tabSelected(tab) {
        this.tabSelectedSubject.next(tab);
    }
}
MdlLayoutMediatorService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutMediatorService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
MdlLayoutMediatorService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutMediatorService, providedIn: "root" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutMediatorService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: "root",
                }]
        }] });
