// @angular-mdl/core/esm2020/lib/layout/mdl-layout-spacer.component.mjs
import { Component, HostBinding, ViewEncapsulation } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlLayoutSpacerComponent {
    constructor() {
        this.isLayoutSpacer = true;
    }
}
MdlLayoutSpacerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutSpacerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlLayoutSpacerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlLayoutSpacerComponent, selector: "mdl-layout-spacer", host: { properties: { "class.mdl-layout-spacer": "this.isLayoutSpacer" } }, ngImport: i0, template: "", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutSpacerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-layout-spacer",
                    template: "",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isLayoutSpacer: [{
                type: HostBinding,
                args: ["class.mdl-layout-spacer"]
            }] } });
