// @angular-mdl/core/esm2020/lib/layout/mdl-layout-tab-panel.component.mjs
import { Component, ContentChild, HostBinding, Input, ViewEncapsulation, } from "@angular/core";
import { MdlTabPanelTitleComponent } from "../tabs/mdl-tab-panel-title.component";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MdlLayoutTabPanelComponent {
    constructor() {
        this.isLayoutTabPanel = true;
        this.isActive = false;
    }
}
MdlLayoutTabPanelComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutTabPanelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlLayoutTabPanelComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlLayoutTabPanelComponent, selector: "mdl-layout-tab-panel", inputs: { title: ["mdl-layout-tab-panel-title", "title"] }, host: { properties: { "class.mdl-layout__tab-panel": "this.isLayoutTabPanel", "class.is-active": "this.isActive" } }, queries: [{ propertyName: "titleComponent", first: true, predicate: MdlTabPanelTitleComponent, descendants: true, static: true }], ngImport: i0, template: `
    <ng-content
      *ngIf="titleComponent"
      select="mdl-tab-panel-content"
    ></ng-content>
    <ng-content *ngIf="!titleComponent"></ng-content>
  `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutTabPanelComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-layout-tab-panel",
                    template: `
    <ng-content
      *ngIf="titleComponent"
      select="mdl-tab-panel-content"
    ></ng-content>
    <ng-content *ngIf="!titleComponent"></ng-content>
  `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { titleComponent: [{
                type: ContentChild,
                args: [MdlTabPanelTitleComponent, { static: true }]
            }], title: [{
                type: Input,
                args: ["mdl-layout-tab-panel-title"]
            }], isLayoutTabPanel: [{
                type: HostBinding,
                args: ["class.mdl-layout__tab-panel"]
            }], isActive: [{
                type: HostBinding,
                args: ["class.is-active"]
            }] } });
