// @angular-mdl/core/esm2020/lib/layout/mdl-layout.module.mjs
import { NgModule } from "@angular/core";
import { MdlLayoutComponent, MdlScreenSizeService, } from "./mdl-layout.component";
import { MdlLayoutHeaderComponent } from "./mdl-layout-header.component";
import { MdlLayoutDrawerComponent } from "./mdl-layout-drawer.component";
import { MdlLayoutContentComponent } from "./mdl-layout-content.component";
import { MdlLayoutHeaderTransparentDirective } from "./mdl-layout-header-transparent.directive";
import { MdlLayoutHeaderRowComponent } from "./mdl-layout-header-row.component";
import { MdlLayoutTitleComponent } from "./mdl-layout-title.component";
import { MdlLayoutSpacerComponent } from "./mdl-layout-spacer.component";
import { MdlLayoutTabPanelComponent } from "./mdl-layout-tab-panel.component";
import { CommonModule } from "@angular/common";
import { MdlIconModule } from "../icon/mdl-icon.module";
import { MdlCommonsModule } from "../common/mdl-common.module";
import { MdlTabsModule } from "../tabs/mdl-tabs.module";
import { MdlRippleModule } from "../ripple/mdl-ripple.module";
import * as i0 from "@angular/core";
const MDL_LAYOUT_DIRECTIVES = [
    MdlLayoutComponent,
    MdlLayoutHeaderComponent,
    MdlLayoutDrawerComponent,
    MdlLayoutContentComponent,
    MdlLayoutHeaderTransparentDirective,
    MdlLayoutHeaderRowComponent,
    MdlLayoutTitleComponent,
    MdlLayoutSpacerComponent,
    MdlLayoutTabPanelComponent,
];
export * from "./mdl-layout.component";
export * from "./mdl-layout-header.component";
export * from "./mdl-layout-drawer.component";
export * from "./mdl-layout-content.component";
export * from "./mdl-layout-header-transparent.directive";
export * from "./mdl-layout-header-row.component";
export * from "./mdl-layout-title.component";
export * from "./mdl-layout-spacer.component";
export * from "./mdl-layout-tab-panel.component";
export class MdlLayoutModule {
    static forRoot() {
        return {
            ngModule: MdlLayoutModule,
            providers: [MdlScreenSizeService],
        };
    }
}
MdlLayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlLayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutModule, declarations: [MdlLayoutComponent,
        MdlLayoutHeaderComponent,
        MdlLayoutDrawerComponent,
        MdlLayoutContentComponent,
        MdlLayoutHeaderTransparentDirective,
        MdlLayoutHeaderRowComponent,
        MdlLayoutTitleComponent,
        MdlLayoutSpacerComponent,
        MdlLayoutTabPanelComponent], imports: [MdlIconModule,
        MdlCommonsModule,
        MdlRippleModule,
        MdlTabsModule,
        CommonModule], exports: [MdlLayoutComponent,
        MdlLayoutHeaderComponent,
        MdlLayoutDrawerComponent,
        MdlLayoutContentComponent,
        MdlLayoutHeaderTransparentDirective,
        MdlLayoutHeaderRowComponent,
        MdlLayoutTitleComponent,
        MdlLayoutSpacerComponent,
        MdlLayoutTabPanelComponent] });
MdlLayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutModule, imports: [MdlIconModule,
        MdlCommonsModule,
        MdlRippleModule,
        MdlTabsModule,
        CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        MdlIconModule,
                        MdlCommonsModule,
                        MdlRippleModule,
                        MdlTabsModule,
                        CommonModule,
                    ],
                    exports: MDL_LAYOUT_DIRECTIVES,
                    declarations: MDL_LAYOUT_DIRECTIVES,
                }]
        }] });
