// @angular-mdl/core/esm2020/lib/layout/mdl-layout-drawer.component.mjs
import { Component, HostBinding, ViewEncapsulation } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlLayoutDrawerComponent {
    constructor() {
        this.isDrawer = true;
        this.isDrawerVisible = false;
    }
}
MdlLayoutDrawerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutDrawerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlLayoutDrawerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlLayoutDrawerComponent, selector: "mdl-layout-drawer", host: { properties: { "class.mdl-layout__drawer": "this.isDrawer", "class.is-visible": "this.isDrawerVisible" } }, ngImport: i0, template: ` <ng-content></ng-content>`, isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutDrawerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-layout-drawer",
                    template: ` <ng-content></ng-content>`,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isDrawer: [{
                type: HostBinding,
                args: ["class.mdl-layout__drawer"]
            }], isDrawerVisible: [{
                type: HostBinding,
                args: ["class.is-visible"]
            }] } });
