// @angular-mdl/core/esm2020/lib/layout/mdl-layout-header-row.component.mjs
import { Component, HostBinding, ViewEncapsulation } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlLayoutHeaderRowComponent {
    constructor() {
        this.isHeaderRow = true;
    }
}
MdlLayoutHeaderRowComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutHeaderRowComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlLayoutHeaderRowComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlLayoutHeaderRowComponent, selector: "mdl-layout-header-row", host: { properties: { "class.mdl-layout__header-row": "this.isHeaderRow" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutHeaderRowComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-layout-header-row",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isHeaderRow: [{
                type: HostBinding,
                args: ["class.mdl-layout__header-row"]
            }] } });
