// @angular-mdl/core/esm2020/lib/layout/mdl-layout-title.component.mjs
import { Component, HostBinding, ViewEncapsulation } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlLayoutTitleComponent {
    constructor() {
        this.isLayoutTitle = true;
    }
}
MdlLayoutTitleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlLayoutTitleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlLayoutTitleComponent, selector: "mdl-layout-title", host: { properties: { "class.mdl-layout-title": "this.isLayoutTitle" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-layout-title",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isLayoutTitle: [{
                type: HostBinding,
                args: ["class.mdl-layout-title"]
            }] } });
