// @angular-mdl/core/esm2020/lib/layout/mdl-layout-content.component.mjs
import { Component, ContentChildren, ElementRef, HostBinding, QueryList, ViewEncapsulation, } from "@angular/core";
import { MdlLayoutTabPanelComponent } from "./mdl-layout-tab-panel.component";
import * as i0 from "@angular/core";
export class MdlLayoutContentComponent {
    constructor(elRef) {
        this.elRef = elRef;
        this.isContent = true;
        this.tabs = new QueryList();
        this.el = elRef.nativeElement;
    }
}
MdlLayoutContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutContentComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
MdlLayoutContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlLayoutContentComponent, selector: "mdl-layout-content", host: { properties: { "class.mdl-layout__content": "this.isContent" } }, queries: [{ propertyName: "tabs", predicate: MdlLayoutTabPanelComponent }], ngImport: i0, template: ` <ng-content></ng-content>`, isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlLayoutContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-layout-content",
                    template: ` <ng-content></ng-content>`,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isContent: [{
                type: HostBinding,
                args: ["class.mdl-layout__content"]
            }], tabs: [{
                type: ContentChildren,
                args: [MdlLayoutTabPanelComponent]
            }] } });
