// @angular-mdl/core/esm2020/lib/progress/mdl-progress.component.mjs
import { ChangeDetectionStrategy, Component, HostBinding, Input, ViewEncapsulation, } from "@angular/core";
import { toBoolean } from "../common/boolean-property";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MdlProgressComponent {
    constructor() {
        this.progress = 0;
        this.buffer = 100;
        this.aux = 0;
        this.isProgess = true;
        this.indeterminateIntern = false;
    }
    get indeterminate() {
        return this.indeterminateIntern;
    }
    set indeterminate(value) {
        this.indeterminateIntern = toBoolean(value);
    }
    ngOnChanges(changes) {
        if (changes["buffer"]) {
            this.setBuffer(changes["buffer"].currentValue);
        }
    }
    setBuffer(b) {
        this.aux = 100 - b;
    }
}
MdlProgressComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlProgressComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlProgressComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlProgressComponent, selector: "mdl-progress", inputs: { progress: "progress", buffer: "buffer", aux: "aux", indeterminate: "indeterminate" }, host: { properties: { "class.mdl-progress": "this.isProgess", "class.mdl-progress__indeterminate": "this.indeterminate" } }, usesOnChanges: true, ngImport: i0, template: `
    <div class="progressbar bar bar1" [style.width]="progress + '%'"></div>
    <div class="bufferbar bar bar2" [style.width]="buffer + '%'"></div>
    <div class="auxbar bar bar3" [ngStyle]="{ width: aux + '%' }"></div>
  `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgStyle, selector: "[ngStyle]", inputs: ["ngStyle"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlProgressComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-progress",
                    template: `
    <div class="progressbar bar bar1" [style.width]="progress + '%'"></div>
    <div class="bufferbar bar bar2" [style.width]="buffer + '%'"></div>
    <div class="auxbar bar bar3" [ngStyle]="{ width: aux + '%' }"></div>
  `,
                    encapsulation: ViewEncapsulation.None,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { progress: [{
                type: Input
            }], buffer: [{
                type: Input
            }], aux: [{
                type: Input
            }], isProgess: [{
                type: HostBinding,
                args: ["class.mdl-progress"]
            }], indeterminate: [{
                type: HostBinding,
                args: ["class.mdl-progress__indeterminate"]
            }, {
                type: Input
            }] } });
