// @angular-mdl/core/esm2020/lib/progress/mdl-progress.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { MdlProgressComponent } from "./mdl-progress.component";
import * as i0 from "@angular/core";
const MDL_PROGRESS_DIRECTIVES = [MdlProgressComponent];
export class MdlProgressModule {
    static forRoot() {
        return {
            ngModule: MdlProgressModule,
            providers: [],
        };
    }
}
MdlProgressModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlProgressModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlProgressModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlProgressModule, declarations: [MdlProgressComponent], imports: [CommonModule], exports: [MdlProgressComponent] });
MdlProgressModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlProgressModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlProgressModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    exports: MDL_PROGRESS_DIRECTIVES,
                    declarations: MDL_PROGRESS_DIRECTIVES,
                }]
        }] });
export * from "./mdl-progress.component";
