// @angular-mdl/core/esm2020/lib/slider/mdl-slider.module.mjs
import { NgModule } from "@angular/core";
import { FormsModule } from "@angular/forms";
import { CommonModule } from "@angular/common";
import { MdlSliderComponent } from "./mdl-slider.component";
import * as i0 from "@angular/core";
const MDL_SLIDER_DIRECTIVES = [MdlSliderComponent];
export class MdlSliderModule {
    static forRoot() {
        return {
            ngModule: MdlSliderModule,
            providers: [],
        };
    }
}
MdlSliderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSliderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlSliderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlSliderModule, declarations: [MdlSliderComponent], imports: [FormsModule, CommonModule], exports: [MdlSliderComponent] });
MdlSliderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSliderModule, imports: [FormsModule, CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSliderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [FormsModule, CommonModule],
                    exports: MDL_SLIDER_DIRECTIVES,
                    declarations: MDL_SLIDER_DIRECTIVES,
                }]
        }] });
export * from "./mdl-slider.component";
