// @angular-mdl/datepicker/esm2020/lib/datepicker.module.mjs
import { NgModule } from "@angular/core";
import { DatePickerDialogComponent } from "./datepicker.component";
import { MdlButtonModule, MdlDialogModule, MdlDialogOutletModule, MdlIconModule, MdlRippleModule, } from "@angular-mdl/core";
import { MdlDatePickerService } from "./datepicker.service";
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
export class MdlDatePickerModule {
}
MdlDatePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlDatePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerModule, declarations: [DatePickerDialogComponent], imports: [CommonModule,
        MdlButtonModule,
        MdlIconModule,
        MdlRippleModule,
        MdlDialogModule,
        MdlDialogOutletModule], exports: [DatePickerDialogComponent] });
MdlDatePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerModule, providers: [MdlDatePickerService], imports: [CommonModule,
        MdlButtonModule,
        MdlIconModule,
        MdlRippleModule,
        MdlDialogModule,
        MdlDialogOutletModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DatePickerDialogComponent],
                    imports: [
                        CommonModule,
                        MdlButtonModule,
                        MdlIconModule,
                        MdlRippleModule,
                        MdlDialogModule,
                        MdlDialogOutletModule,
                    ],
                    exports: [DatePickerDialogComponent],
                    providers: [MdlDatePickerService],
                }]
        }] });
