// @angular-mdl/datepicker/esm2020/lib/datepicker.service.mjs
import { Injectable } from "@angular/core";
import { Subject } from "rxjs";
import { MdlDialogService } from "@angular-mdl/core";
import { CURRENT_DATE, DATEPICKER_CONFIG, DatePickerDialogComponent, } from "./datepicker.component";
import * as i0 from "@angular/core";
import * as i1 from "@angular-mdl/core";
export class MdlDatePickerService {
    constructor(dialogService) {
        this.dialogService = dialogService;
    }
    selectDate(currentDate = null, options = {}) {
        const subject = new Subject();
        const pDialog = this.dialogService.showCustomDialog({
            classes: "mdl-datepicker",
            component: DatePickerDialogComponent,
            providers: [
                { provide: CURRENT_DATE, useValue: currentDate },
                { provide: DATEPICKER_CONFIG, useValue: options },
            ],
            isModal: true,
            styles: { width: "320px" },
            openFrom: options.openFrom,
        });
        pDialog.subscribe((dialogReference) => {
            dialogReference.onHide().subscribe((date) => {
                subject.next(date);
                subject.complete();
            });
        });
        return subject.asObservable();
    }
}
MdlDatePickerService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerService, deps: [{ token: i1.MdlDialogService }], target: i0.ɵɵFactoryTarget.Injectable });
MdlDatePickerService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDatePickerService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MdlDialogService }]; } });
