// @angular-mdl/core/esm2020/lib/dialog-outlet/mdl-dialog-outlet.component.mjs
import { Component, ViewContainerRef } from "@angular/core";
import * as i0 from "@angular/core";
// the componnet is used outside the app-root. injecting MdlDialogService would not work
// this component is not exported - needs to be instanciated by
//    let x = this.appRef.bootstrap(MdlDialogOutletComponent);
export class MdlDialogOutletComponent {
    constructor(vCRef) {
        this.vCRef = vCRef;
    }
    get viewContainerRef() {
        return this.vCRef;
    }
}
MdlDialogOutletComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletComponent, deps: [{ token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Component });
MdlDialogOutletComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlDialogOutletComponent, selector: "dialog-outlet", ngImport: i0, template: "", isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletComponent, decorators: [{
            type: Component,
            args: [{
                    // eslint-disable-next-line
                    selector: 'dialog-outlet',
                    template: "",
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }]; } });
