// @angular-mdl/core/esm2020/lib/dialog-outlet/mdl-dialog-outlet.module.mjs
import { NgModule } from "@angular/core";
import { MdlDialogOutletComponent } from "./mdl-dialog-outlet.component";
import { MdlDialogOutletService } from "./mdl-dialog-outlet.service";
import { MdlBackdropOverlayComponent } from "./mdl-backdrop-overlay.component";
import { MdlDialogInnerOutletComponent } from "./mdl-dialog-inner-outlet.component";
import * as i0 from "@angular/core";
export * from "./mdl-dialog-outlet.component";
export * from "./mdl-dialog-outlet.service";
export * from "./mdl-backdrop-overlay.component";
export * from "./mdl-dialog-inner-outlet.component";
const PUBLIC_COMPONENTS = [MdlDialogInnerOutletComponent];
const PRIVATE_COMPONENTS = [
    MdlDialogOutletComponent,
    MdlBackdropOverlayComponent,
];
export class MdlDialogOutletModule {
    static forRoot() {
        return {
            ngModule: MdlDialogOutletModule,
            providers: [MdlDialogOutletService],
        };
    }
}
MdlDialogOutletModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlDialogOutletModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletModule, declarations: [MdlDialogInnerOutletComponent, MdlDialogOutletComponent,
        MdlBackdropOverlayComponent], exports: [MdlDialogInnerOutletComponent] });
MdlDialogOutletModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: PUBLIC_COMPONENTS,
                    declarations: [...PUBLIC_COMPONENTS, ...PRIVATE_COMPONENTS],
                }]
        }] });
