// @angular-mdl/core/esm2020/lib/dialog-outlet/mdl-dialog-inner-outlet.component.mjs
// the component is used inside the app-root. this is possible because this component
// is exported from the module
import { Component, forwardRef, Inject, ViewContainerRef } from "@angular/core";
import { MdlDialogOutletService } from "./mdl-dialog-outlet.service";
import * as i0 from "@angular/core";
import * as i1 from "./mdl-dialog-outlet.service";
export class MdlDialogInnerOutletComponent {
    constructor(vCRef, service) {
        this.vCRef = vCRef;
        service.setDefaultViewContainerRef(vCRef);
    }
}
MdlDialogInnerOutletComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogInnerOutletComponent, deps: [{ token: i0.ViewContainerRef }, { token: forwardRef(() => MdlDialogOutletService) }], target: i0.ɵɵFactoryTarget.Component });
MdlDialogInnerOutletComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlDialogInnerOutletComponent, selector: "dialog-outlet", ngImport: i0, template: "", isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogInnerOutletComponent, decorators: [{
            type: Component,
            args: [{
                    // eslint-disable-next-line
                    selector: 'dialog-outlet',
                    template: "",
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i1.MdlDialogOutletService, decorators: [{
                    type: Inject,
                    args: [forwardRef(() => MdlDialogOutletService)]
                }] }]; } });
