// @angular-mdl/core/esm2020/lib/dialog-outlet/mdl-dialog-outlet.service.mjs
import { ApplicationRef, ComponentFactoryResolver, EventEmitter, Injectable, } from "@angular/core";
import { MdlBackdropOverlayComponent } from "./mdl-backdrop-overlay.component";
import { filter, take } from "rxjs/operators";
import { MdlDialogOutletComponent } from "./mdl-dialog-outlet.component";
import * as i0 from "@angular/core";
export class MdlDialogOutletService {
    constructor(appRef, componentFactoryResolver) {
        this.appRef = appRef;
        this.componentFactoryResolver = componentFactoryResolver;
        this.backdropClickEmitter = new EventEmitter();
        this.viewContainerRefInternal = null;
        let dialogOutletCompRef = null;
        appRef.isStable
            .pipe(take(1), filter(() => this.viewContainerRefInternal == null))
            .subscribe(() => {
            try {
                dialogOutletCompRef = this.appRef.bootstrap(MdlDialogOutletComponent);
            }
            catch (e) {
                // the user did not use the dialog.outlet element outside of his root app.
                // console.log(e);
            }
            if (dialogOutletCompRef) {
                this.setViewContainerRef(dialogOutletCompRef.instance.viewContainerRef);
            }
        });
    }
    get viewContainerRef() {
        return this.viewContainerRefInternal;
    }
    setDefaultViewContainerRef(vCRef) {
        this.setViewContainerRef(vCRef);
    }
    hideBackdrop() {
        this.backdropComponent?.hide();
    }
    showBackdropWithZIndex(zIndex) {
        this.backdropComponent?.showWithZIndex(zIndex);
    }
    setViewContainerRef(value) {
        this.viewContainerRefInternal = value;
        if (this.viewContainerRefInternal) {
            const cFactory = this.componentFactoryResolver.resolveComponentFactory(MdlBackdropOverlayComponent);
            this.backdropComponent =
                this.viewContainerRefInternal.createComponent(cFactory).instance;
            this.backdropComponent.clickEmitter.subscribe(() => {
                this.backdropClickEmitter.emit();
            });
        }
    }
}
MdlDialogOutletService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletService, deps: [{ token: i0.ApplicationRef }, { token: i0.ComponentFactoryResolver }], target: i0.ɵɵFactoryTarget.Injectable });
MdlDialogOutletService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletService, providedIn: "root" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogOutletService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: "root",
                }]
        }], ctorParameters: function () { return [{ type: i0.ApplicationRef }, { type: i0.ComponentFactoryResolver }]; } });
