// @angular-mdl/core/esm2020/lib/dialog-outlet/mdl-backdrop-overlay.component.mjs
import { Component, EventEmitter, HostBinding, HostListener, NgZone, ViewEncapsulation, } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlBackdropOverlayComponent {
    constructor(ngZone) {
        this.ngZone = ngZone;
        this.zIndex = 0;
        this.isBackdrop = true;
        this.clickEmitter = new EventEmitter();
        this.visible = false;
    }
    get display() {
        return this.visible ? null : "none";
    }
    onBackdropClick(e) {
        // this event runs not in angular zone of the main app. make sure it runs in the main angular zone
        // and change detection works
        this.ngZone.run(() => {
            this.clickEmitter.emit();
        });
        e.stopPropagation();
    }
    hide() {
        this.visible = false;
    }
    showWithZIndex(zIndex) {
        this.zIndex = zIndex;
        this.visible = true;
    }
}
MdlBackdropOverlayComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBackdropOverlayComponent, deps: [{ token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component });
MdlBackdropOverlayComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlBackdropOverlayComponent, selector: "mdl-backdrop-overlay", host: { listeners: { "click": "onBackdropClick($event)" }, properties: { "style.zIndex": "this.zIndex", "class.dialog-backdrop": "this.isBackdrop", "style.display": "this.display" } }, ngImport: i0, template: ``, isInline: true, styles: [".dialog-backdrop{position:fixed;inset:0;background:rgba(0,0,0,.1)}\n"], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBackdropOverlayComponent, decorators: [{
            type: Component,
            args: [{ selector: "mdl-backdrop-overlay", template: ``, encapsulation: ViewEncapsulation.None, styles: [".dialog-backdrop{position:fixed;inset:0;background:rgba(0,0,0,.1)}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.NgZone }]; }, propDecorators: { zIndex: [{
                type: HostBinding,
                args: ["style.zIndex"]
            }], isBackdrop: [{
                type: HostBinding,
                args: ["class.dialog-backdrop"]
            }], display: [{
                type: HostBinding,
                args: ["style.display"]
            }], onBackdropClick: [{
                type: HostListener,
                args: ["click", ["$event"]]
            }] } });
