// @angular-mdl/core/esm2020/lib/textfield/mdl-textfield.module.mjs
import { NgModule } from "@angular/core";
import { FormsModule } from "@angular/forms";
import { CommonModule } from "@angular/common";
import { MdlTextFieldComponent } from "./mdl-textfield.component";
import { MdlIconModule } from "../icon/mdl-icon.module";
import { MdlButtonModule } from "../button/mdl-button.module";
import * as i0 from "@angular/core";
export class MdlTextFieldModule {
    static forRoot() {
        return {
            ngModule: MdlTextFieldModule,
            providers: [],
        };
    }
}
MdlTextFieldModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTextFieldModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlTextFieldModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlTextFieldModule, declarations: [MdlTextFieldComponent], imports: [MdlIconModule, MdlButtonModule, FormsModule, CommonModule], exports: [MdlTextFieldComponent] });
MdlTextFieldModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTextFieldModule, imports: [MdlIconModule, MdlButtonModule, FormsModule, CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTextFieldModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [MdlIconModule, MdlButtonModule, FormsModule, CommonModule],
                    exports: [MdlTextFieldComponent],
                    declarations: [MdlTextFieldComponent],
                }]
        }] });
export * from "./mdl-textfield.component";
