// @angular-mdl/core/esm2020/lib/card/mdl-card.module.mjs
import { NgModule } from "@angular/core";
import { MdlCardActionsComponent, MdlCardBorderDirective, MdlCardComponent, MdlCardExpandDirective, MdlCardMediaComponent, MdlCardMenuComponent, MdlCardSupportingTextComponent, MdlCardTitleComponent, MdlCardTitleTextDirective, } from "./mdl-card.component";
import * as i0 from "@angular/core";
const MDL_CARD_DIRECTIVES = [
    MdlCardComponent,
    MdlCardTitleComponent,
    MdlCardMediaComponent,
    MdlCardSupportingTextComponent,
    MdlCardActionsComponent,
    MdlCardMenuComponent,
    MdlCardTitleTextDirective,
    MdlCardBorderDirective,
    MdlCardExpandDirective,
];
export class MdlCardModule {
    static forRoot() {
        return {
            ngModule: MdlCardModule,
            providers: [],
        };
    }
}
MdlCardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlCardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlCardModule, declarations: [MdlCardComponent,
        MdlCardTitleComponent,
        MdlCardMediaComponent,
        MdlCardSupportingTextComponent,
        MdlCardActionsComponent,
        MdlCardMenuComponent,
        MdlCardTitleTextDirective,
        MdlCardBorderDirective,
        MdlCardExpandDirective], exports: [MdlCardComponent,
        MdlCardTitleComponent,
        MdlCardMediaComponent,
        MdlCardSupportingTextComponent,
        MdlCardActionsComponent,
        MdlCardMenuComponent,
        MdlCardTitleTextDirective,
        MdlCardBorderDirective,
        MdlCardExpandDirective] });
MdlCardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_CARD_DIRECTIVES,
                    declarations: MDL_CARD_DIRECTIVES,
                }]
        }] });
export * from "./mdl-card.component";
