// @angular-mdl/core/esm2020/lib/card/mdl-card.component.mjs
import { Component, Directive, HostBinding, Optional, ViewEncapsulation, } from "@angular/core";
import { MdlStructureError } from "../common/mdl-error";
import * as i0 from "@angular/core";
export class MdlCardComponent {
    constructor() {
        this.isCard = true;
    }
}
MdlCardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlCardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardComponent, selector: "mdl-card", host: { properties: { "class.mdl-card": "this.isCard" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-card",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isCard: [{
                type: HostBinding,
                args: ["class.mdl-card"]
            }] } });
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class MdlCardChildStructure {
    constructor(mdlCardComponent, childComponentName) {
        this.mdlCardComponent = mdlCardComponent;
        this.childComponentName = childComponentName;
    }
    ngOnInit() {
        if (this.mdlCardComponent === null) {
            throw new MdlStructureError(this.childComponentName, "mdl-card");
        }
    }
}
MdlCardChildStructure.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardChildStructure, deps: "invalid", target: i0.ɵɵFactoryTarget.Directive });
MdlCardChildStructure.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardChildStructure, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardChildStructure, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: MdlCardComponent }, { type: undefined }]; } });
export class MdlCardTitleComponent extends MdlCardChildStructure {
    constructor(mdlCardComponent) {
        super(mdlCardComponent, "mdl-card-title");
        this.isCardTitle = true;
    }
}
MdlCardTitleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardTitleComponent, deps: [{ token: MdlCardComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlCardTitleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardTitleComponent, selector: "mdl-card-title", host: { properties: { "class.mdl-card__title": "this.isCardTitle" } }, usesInheritance: true, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-card-title",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlCardComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isCardTitle: [{
                type: HostBinding,
                args: ["class.mdl-card__title"]
            }] } });
export class MdlCardSupportingTextComponent extends MdlCardChildStructure {
    constructor(mdlCardComponent) {
        super(mdlCardComponent, "mdl-card-supporting-text");
        this.isSupportingText = true;
    }
}
MdlCardSupportingTextComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardSupportingTextComponent, deps: [{ token: MdlCardComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlCardSupportingTextComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardSupportingTextComponent, selector: "mdl-card-supporting-text", host: { properties: { "class.mdl-card__supporting-text": "this.isSupportingText" } }, usesInheritance: true, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardSupportingTextComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-card-supporting-text",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlCardComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isSupportingText: [{
                type: HostBinding,
                args: ["class.mdl-card__supporting-text"]
            }] } });
export class MdlCardMediaComponent extends MdlCardChildStructure {
    constructor(mdlCardComponent) {
        super(mdlCardComponent, "mdl-card-media");
        this.isCardMedia = true;
    }
}
MdlCardMediaComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardMediaComponent, deps: [{ token: MdlCardComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlCardMediaComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardMediaComponent, selector: "mdl-card-media", host: { properties: { "class.mdl-card__media": "this.isCardMedia" } }, usesInheritance: true, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardMediaComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-card-media",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlCardComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isCardMedia: [{
                type: HostBinding,
                args: ["class.mdl-card__media"]
            }] } });
export class MdlCardActionsComponent extends MdlCardChildStructure {
    constructor(mdlCardComponent) {
        super(mdlCardComponent, "mdl-card-actions");
        this.isCardAction = true;
    }
}
MdlCardActionsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardActionsComponent, deps: [{ token: MdlCardComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlCardActionsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardActionsComponent, selector: "mdl-card-actions", host: { properties: { "class.mdl-card__actions": "this.isCardAction" } }, usesInheritance: true, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardActionsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-card-actions",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlCardComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isCardAction: [{
                type: HostBinding,
                args: ["class.mdl-card__actions"]
            }] } });
export class MdlCardMenuComponent extends MdlCardChildStructure {
    constructor(mdlCardComponent) {
        super(mdlCardComponent, "mdl-card-menu");
        this.isCardMenu = true;
    }
}
MdlCardMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardMenuComponent, deps: [{ token: MdlCardComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlCardMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardMenuComponent, selector: "mdl-card-menu", host: { properties: { "class.mdl-card__menu": "this.isCardMenu" } }, usesInheritance: true, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-card-menu",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlCardComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isCardMenu: [{
                type: HostBinding,
                args: ["class.mdl-card__menu"]
            }] } });
export class MdlCardTitleTextDirective {
    constructor() {
        this.isCardTitleText = true;
    }
}
MdlCardTitleTextDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardTitleTextDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlCardTitleTextDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardTitleTextDirective, selector: "[mdl-card-title-text]", host: { properties: { "class.mdl-card__title-text": "this.isCardTitleText" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardTitleTextDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: '[mdl-card-title-text]'
                }]
        }], propDecorators: { isCardTitleText: [{
                type: HostBinding,
                args: ["class.mdl-card__title-text"]
            }] } });
export class MdlCardBorderDirective {
    constructor() {
        this.isCardBorder = true;
    }
}
MdlCardBorderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardBorderDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlCardBorderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardBorderDirective, selector: "[mdl-card-border]", host: { properties: { "class.mdl-card--border": "this.isCardBorder" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardBorderDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: '[mdl-card-border]'
                }]
        }], propDecorators: { isCardBorder: [{
                type: HostBinding,
                args: ["class.mdl-card--border"]
            }] } });
export class MdlCardExpandDirective {
    constructor() {
        this.isCardExpand = true;
    }
}
MdlCardExpandDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardExpandDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlCardExpandDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlCardExpandDirective, selector: "[mdl-card-expand]", host: { properties: { "class.mdl-card--expand": "this.isCardExpand" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCardExpandDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: '[mdl-card-expand]'
                }]
        }], propDecorators: { isCardExpand: [{
                type: HostBinding,
                args: ["class.mdl-card--expand"]
            }] } });
