// @angular-mdl/core/esm2020/lib/ripple/mdl-ripple.module.mjs
import { NgModule } from "@angular/core";
import { MdlAnchorRippleDirective, MdlButtonRippleDirective, MdlCheckboxRippleDirective, MdlIconToggleRippleDirective, MdlMenuItemRippleDirective, MdlRadioRippleDirective, MdlSwitchRippleDirective, } from "./mdl-ripple.directive";
import * as i0 from "@angular/core";
const MDL_RIPPLE_DIRECTIVES = [
    MdlCheckboxRippleDirective,
    MdlButtonRippleDirective,
    MdlRadioRippleDirective,
    MdlIconToggleRippleDirective,
    MdlSwitchRippleDirective,
    MdlMenuItemRippleDirective,
    MdlAnchorRippleDirective,
];
export class MdlRippleModule {
    static forRoot() {
        return {
            ngModule: MdlRippleModule,
            providers: [],
        };
    }
}
MdlRippleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlRippleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlRippleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlRippleModule, declarations: [MdlCheckboxRippleDirective,
        MdlButtonRippleDirective,
        MdlRadioRippleDirective,
        MdlIconToggleRippleDirective,
        MdlSwitchRippleDirective,
        MdlMenuItemRippleDirective,
        MdlAnchorRippleDirective], exports: [MdlCheckboxRippleDirective,
        MdlButtonRippleDirective,
        MdlRadioRippleDirective,
        MdlIconToggleRippleDirective,
        MdlSwitchRippleDirective,
        MdlMenuItemRippleDirective,
        MdlAnchorRippleDirective] });
MdlRippleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlRippleModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlRippleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_RIPPLE_DIRECTIVES,
                    declarations: MDL_RIPPLE_DIRECTIVES,
                }]
        }] });
export * from "../ripple/mdl-ripple.directive";
