// @angular-mdl/core/esm2020/lib/radio/mdl-radio.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { FormsModule } from "@angular/forms";
import { MdlRadioComponent, MdlRadioGroupRegisty } from "./mdl-radio.component";
import * as i0 from "@angular/core";
export class MdlRadioModule {
    static forRoot() {
        return {
            ngModule: MdlRadioModule,
            providers: [MdlRadioGroupRegisty],
        };
    }
}
MdlRadioModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlRadioModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlRadioModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlRadioModule, declarations: [MdlRadioComponent], imports: [CommonModule, FormsModule], exports: [MdlRadioComponent] });
MdlRadioModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlRadioModule, imports: [CommonModule, FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlRadioModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule],
                    exports: [MdlRadioComponent],
                    declarations: [MdlRadioComponent],
                }]
        }] });
export * from "./mdl-radio.component";
