// @angular-mdl/core/esm2020/lib/badge/mdl-badge.directive.mjs
import { Directive, ElementRef, HostBinding, Input, Renderer2, } from "@angular/core";
import * as i0 from "@angular/core";
const DATA_BADE_ATTR = "data-badge";
export class MdlBadgeDirective {
    constructor(elementRef, renderer) {
        this.elementRef = elementRef;
        this.renderer = renderer;
        this.isBadge = true;
        this.el = elementRef.nativeElement;
    }
    ngOnChanges() {
        if (this.mdlBadgeContent === null ||
            typeof this.mdlBadgeContent === "undefined") {
            this.renderer.removeAttribute(this.el, DATA_BADE_ATTR);
            return;
        }
        this.renderer.setAttribute(this.el, DATA_BADE_ATTR, String(this.mdlBadgeContent));
    }
}
MdlBadgeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeDirective, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
MdlBadgeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlBadgeDirective, selector: "[mdl-badge]", inputs: { mdlBadgeContent: ["mdl-badge", "mdlBadgeContent"] }, host: { properties: { "class.mdl-badge": "this.isBadge" } }, usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "[mdl-badge]",
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }]; }, propDecorators: { mdlBadgeContent: [{
                type: Input,
                args: ["mdl-badge"]
            }], isBadge: [{
                type: HostBinding,
                args: ["class.mdl-badge"]
            }] } });
export class MdlBadgeOverlapDirective {
    constructor() {
        this.isOverlapping = true;
    }
}
MdlBadgeOverlapDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeOverlapDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlBadgeOverlapDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlBadgeOverlapDirective, selector: "[mdl-badge-overlap]", host: { properties: { "class.mdl-badge--overlap": "this.isOverlapping" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeOverlapDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "[mdl-badge-overlap]",
                }]
        }], propDecorators: { isOverlapping: [{
                type: HostBinding,
                args: ["class.mdl-badge--overlap"]
            }] } });
export class MdlBadgeNoBackgroundDirective {
    constructor() {
        this.isNoBackground = true;
    }
}
MdlBadgeNoBackgroundDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeNoBackgroundDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlBadgeNoBackgroundDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlBadgeNoBackgroundDirective, selector: "[mdl-badge-no-background]", host: { properties: { "class.mdl-badge--no-background": "this.isNoBackground" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeNoBackgroundDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "[mdl-badge-no-background]",
                }]
        }], propDecorators: { isNoBackground: [{
                type: HostBinding,
                args: ["class.mdl-badge--no-background"]
            }] } });
