// @angular-mdl/core/esm2020/lib/badge/mdl-badge.module.mjs
import { NgModule } from "@angular/core";
import { MdlBadgeDirective, MdlBadgeNoBackgroundDirective, MdlBadgeOverlapDirective, } from "./mdl-badge.directive";
import * as i0 from "@angular/core";
const MDL_BADGE_DIRECTIVES = [
    MdlBadgeDirective,
    MdlBadgeOverlapDirective,
    MdlBadgeNoBackgroundDirective,
];
export class MdlBadgeModule {
    static forRoot() {
        return {
            ngModule: MdlBadgeModule,
            providers: [],
        };
    }
}
MdlBadgeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlBadgeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeModule, declarations: [MdlBadgeDirective,
        MdlBadgeOverlapDirective,
        MdlBadgeNoBackgroundDirective], exports: [MdlBadgeDirective,
        MdlBadgeOverlapDirective,
        MdlBadgeNoBackgroundDirective] });
MdlBadgeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlBadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_BADGE_DIRECTIVES,
                    declarations: MDL_BADGE_DIRECTIVES,
                }]
        }] });
export * from "./mdl-badge.directive";
