// @angular-mdl/core/esm2020/lib/button/mdl-button.module.mjs
import { NgModule } from "@angular/core";
import { MdlButtonComponent } from "./mdl-button.component";
import * as i0 from "@angular/core";
const MDL_BUTTON_DIRECTIVES = [MdlButtonComponent];
export class MdlButtonModule {
    static forRoot() {
        return {
            ngModule: MdlButtonModule,
            providers: [],
        };
    }
}
MdlButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlButtonModule, declarations: [MdlButtonComponent], exports: [MdlButtonComponent] });
MdlButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlButtonModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_BUTTON_DIRECTIVES,
                    declarations: MDL_BUTTON_DIRECTIVES,
                }]
        }] });
export * from "./mdl-button.component";
