// @angular-mdl/core/esm2020/lib/menu/mdl-toggle-menu.directive.mjs
import { Directive, HostListener, Input } from "@angular/core";
import { MdlButtonComponent } from "../button/mdl-button.component";
import * as i0 from "@angular/core";
import * as i1 from "../button/mdl-button.component";
export class MdlToggleMenuDirective {
    constructor(button) {
        this.button = button;
    }
    onClick($event) {
        this.menu?.toggle($event, this.button);
    }
}
MdlToggleMenuDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlToggleMenuDirective, deps: [{ token: i1.MdlButtonComponent }], target: i0.ɵɵFactoryTarget.Directive });
MdlToggleMenuDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlToggleMenuDirective, selector: "[mdl-button][mdl-toggle-menu]", inputs: { menu: ["mdl-toggle-menu", "menu"] }, host: { listeners: { "click": "onClick($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlToggleMenuDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "[mdl-button][mdl-toggle-menu]",
                }]
        }], ctorParameters: function () { return [{ type: i1.MdlButtonComponent }]; }, propDecorators: { menu: [{
                type: Input,
                args: ["mdl-toggle-menu"]
            }], onClick: [{
                type: HostListener,
                args: ["click", ["$event"]]
            }] } });
