// @angular-mdl/core/esm2020/lib/menu/mdl-menu.module.mjs
import { NgModule } from "@angular/core";
import { MdlMenuComponent, MdlMenuRegisty } from "./mdl-menu.component";
import { MdlMenuItemComponent } from "./mdl-menu-item.component";
import { MdlMenuItemFullBleedDeviderDirective } from "./mdl-menu-item.directive";
import { CommonModule } from "@angular/common";
import { MdlToggleMenuDirective } from "./mdl-toggle-menu.directive";
import * as i0 from "@angular/core";
const MDL_MENU_DIRECTIVES = [
    MdlMenuComponent,
    MdlMenuItemComponent,
    MdlMenuItemFullBleedDeviderDirective,
    MdlToggleMenuDirective,
];
export * from "./mdl-menu.component";
export * from "./mdl-menu-item.component";
export * from "./mdl-menu-item.directive";
export * from "./mdl-toggle-menu.directive";
export class MdlMenuModule {
    static forRoot() {
        return {
            ngModule: MdlMenuModule,
            providers: [MdlMenuRegisty],
        };
    }
}
MdlMenuModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlMenuModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlMenuModule, declarations: [MdlMenuComponent,
        MdlMenuItemComponent,
        MdlMenuItemFullBleedDeviderDirective,
        MdlToggleMenuDirective], imports: [CommonModule], exports: [MdlMenuComponent,
        MdlMenuItemComponent,
        MdlMenuItemFullBleedDeviderDirective,
        MdlToggleMenuDirective] });
MdlMenuModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlMenuModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    exports: MDL_MENU_DIRECTIVES,
                    declarations: MDL_MENU_DIRECTIVES,
                }]
        }] });
