// @angular-mdl/core/esm2020/lib/core.module.mjs
import { NgModule } from "@angular/core";
import { MdlCommonsModule } from "./common/mdl-common.module";
import { MdlLayoutModule } from "./layout/mdl-layout.module";
import { MdlIconModule } from "./icon/mdl-icon.module";
import { MdlBadgeModule } from "./badge/mdl-badge.module";
import { MdlButtonModule } from "./button/mdl-button.module";
import { MdlCardModule } from "./card/mdl-card.module";
import { MdlCheckboxModule } from "./checkbox/mdl-checkbox.module";
import { MdlChipModule } from "./chips/mdl-chip.module";
import { MdlDialogModule } from "./dialog/mdl-dialog.module";
import { MdlDialogOutletModule } from "./dialog-outlet/mdl-dialog-outlet.module";
import { MdlIconToggleModule } from "./icon-toggle/mdl-icon-toggle.module";
import { MdlListModule } from "./list/mdl-list.module";
import { MdlMenuModule } from "./menu/mdl-menu.module";
import { MdlProgressModule } from "./progress/mdl-progress.module";
import { MdlRadioModule } from "./radio/mdl-radio.module";
import { MdlShadowModule } from "./shadow/mdl-shadow.module";
import { MdlSliderModule } from "./slider/mdl-slider.module";
import { MdlSnackbarModule } from "./snackbar/mdl-snackbar.module";
import { MdlSpinnerModule } from "./spinner/mdl-spinner.module";
import { MdlSwitchModule } from "./switch/mdl-switch.module";
import { MdlTableModule } from "./table/mdl-table.module";
import { MdlTabsModule } from "./tabs/mdl-tabs.module";
import { MdlTextFieldModule } from "./textfield/mdl-textfield.module";
import { MdlTooltipModule } from "./tooltip/mdl-tooltip.module";
import { MdlRippleModule } from "./ripple/mdl-ripple.module";
import * as i0 from "@angular/core";
import * as i1 from "./button/mdl-button.module";
import * as i2 from "./layout/mdl-layout.module";
import * as i3 from "./checkbox/mdl-checkbox.module";
import * as i4 from "./chips/mdl-chip.module";
import * as i5 from "./dialog/mdl-dialog.module";
import * as i6 from "./dialog-outlet/mdl-dialog-outlet.module";
import * as i7 from "./spinner/mdl-spinner.module";
import * as i8 from "./badge/mdl-badge.module";
import * as i9 from "./shadow/mdl-shadow.module";
import * as i10 from "./card/mdl-card.module";
import * as i11 from "./radio/mdl-radio.module";
import * as i12 from "./ripple/mdl-ripple.module";
import * as i13 from "./progress/mdl-progress.module";
import * as i14 from "./icon/mdl-icon.module";
import * as i15 from "./icon-toggle/mdl-icon-toggle.module";
import * as i16 from "./list/mdl-list.module";
import * as i17 from "./slider/mdl-slider.module";
import * as i18 from "./switch/mdl-switch.module";
import * as i19 from "./snackbar/mdl-snackbar.module";
import * as i20 from "./tooltip/mdl-tooltip.module";
import * as i21 from "./table/mdl-table.module";
import * as i22 from "./menu/mdl-menu.module";
import * as i23 from "./tabs/mdl-tabs.module";
import * as i24 from "./textfield/mdl-textfield.module";
const MODULES = [
    MdlBadgeModule,
    MdlButtonModule,
    MdlCardModule,
    MdlCheckboxModule,
    MdlChipModule,
    MdlCommonsModule,
    MdlDialogModule,
    MdlDialogOutletModule,
    MdlIconModule,
    MdlIconToggleModule,
    MdlLayoutModule,
    MdlListModule,
    MdlMenuModule,
    MdlProgressModule,
    MdlRadioModule,
    MdlRippleModule,
    MdlShadowModule,
    MdlSliderModule,
    MdlSnackbarModule,
    MdlSpinnerModule,
    MdlSwitchModule,
    MdlTableModule,
    MdlTabsModule,
    MdlTextFieldModule,
    MdlTooltipModule,
];
export class MdlModule {
}
MdlModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlModule, imports: [i1.MdlButtonModule, i2.MdlLayoutModule, i3.MdlCheckboxModule, i4.MdlChipModule, i5.MdlDialogModule, i6.MdlDialogOutletModule, i7.MdlSpinnerModule, i8.MdlBadgeModule, i9.MdlShadowModule, i10.MdlCardModule, i11.MdlRadioModule, i12.MdlRippleModule, i13.MdlProgressModule, i14.MdlIconModule, i15.MdlIconToggleModule, i16.MdlListModule, i17.MdlSliderModule, i18.MdlSwitchModule, i19.MdlSnackbarModule, i20.MdlTooltipModule, i21.MdlTableModule, i22.MdlMenuModule, i23.MdlTabsModule, i24.MdlTextFieldModule], exports: [MdlBadgeModule,
        MdlButtonModule,
        MdlCardModule,
        MdlCheckboxModule,
        MdlChipModule,
        MdlCommonsModule,
        MdlDialogModule,
        MdlDialogOutletModule,
        MdlIconModule,
        MdlIconToggleModule,
        MdlLayoutModule,
        MdlListModule,
        MdlMenuModule,
        MdlProgressModule,
        MdlRadioModule,
        MdlRippleModule,
        MdlShadowModule,
        MdlSliderModule,
        MdlSnackbarModule,
        MdlSpinnerModule,
        MdlSwitchModule,
        MdlTableModule,
        MdlTabsModule,
        MdlTextFieldModule,
        MdlTooltipModule] });
MdlModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlModule, imports: [MdlButtonModule.forRoot(),
        MdlLayoutModule.forRoot(),
        MdlCheckboxModule.forRoot(),
        MdlChipModule.forRoot(),
        MdlDialogModule.forRoot(),
        MdlDialogOutletModule.forRoot(),
        MdlSpinnerModule.forRoot(),
        MdlBadgeModule.forRoot(),
        MdlShadowModule.forRoot(),
        MdlCardModule.forRoot(),
        MdlRadioModule.forRoot(),
        MdlRippleModule.forRoot(),
        MdlProgressModule.forRoot(),
        MdlIconModule.forRoot(),
        MdlIconToggleModule.forRoot(),
        MdlListModule.forRoot(),
        MdlSliderModule.forRoot(),
        MdlSwitchModule.forRoot(),
        MdlSnackbarModule.forRoot(),
        MdlTooltipModule.forRoot(),
        MdlTableModule.forRoot(),
        MdlMenuModule.forRoot(),
        MdlTabsModule.forRoot(),
        MdlTextFieldModule.forRoot(), MdlBadgeModule,
        MdlButtonModule,
        MdlCardModule,
        MdlCheckboxModule,
        MdlChipModule,
        MdlCommonsModule,
        MdlDialogModule,
        MdlDialogOutletModule,
        MdlIconModule,
        MdlIconToggleModule,
        MdlLayoutModule,
        MdlListModule,
        MdlMenuModule,
        MdlProgressModule,
        MdlRadioModule,
        MdlRippleModule,
        MdlShadowModule,
        MdlSliderModule,
        MdlSnackbarModule,
        MdlSpinnerModule,
        MdlSwitchModule,
        MdlTableModule,
        MdlTabsModule,
        MdlTextFieldModule,
        MdlTooltipModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [
                        MdlButtonModule.forRoot(),
                        MdlLayoutModule.forRoot(),
                        MdlCheckboxModule.forRoot(),
                        MdlChipModule.forRoot(),
                        MdlDialogModule.forRoot(),
                        MdlDialogOutletModule.forRoot(),
                        MdlSpinnerModule.forRoot(),
                        MdlBadgeModule.forRoot(),
                        MdlShadowModule.forRoot(),
                        MdlCardModule.forRoot(),
                        MdlRadioModule.forRoot(),
                        MdlRippleModule.forRoot(),
                        MdlProgressModule.forRoot(),
                        MdlIconModule.forRoot(),
                        MdlIconToggleModule.forRoot(),
                        MdlListModule.forRoot(),
                        MdlSliderModule.forRoot(),
                        MdlSwitchModule.forRoot(),
                        MdlSnackbarModule.forRoot(),
                        MdlTooltipModule.forRoot(),
                        MdlTableModule.forRoot(),
                        MdlMenuModule.forRoot(),
                        MdlTabsModule.forRoot(),
                        MdlTextFieldModule.forRoot(),
                    ],
                    exports: [...MODULES],
                }]
        }] });
// @deprectaed use MdlModule - all services are provided in root by default.
export class MdlNonRootModule {
    static forRoot() {
        return { ngModule: MdlModule };
    }
}
MdlNonRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlNonRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlNonRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlNonRootModule, imports: [MdlBadgeModule,
        MdlButtonModule,
        MdlCardModule,
        MdlCheckboxModule,
        MdlChipModule,
        MdlCommonsModule,
        MdlDialogModule,
        MdlDialogOutletModule,
        MdlIconModule,
        MdlIconToggleModule,
        MdlLayoutModule,
        MdlListModule,
        MdlMenuModule,
        MdlProgressModule,
        MdlRadioModule,
        MdlRippleModule,
        MdlShadowModule,
        MdlSliderModule,
        MdlSnackbarModule,
        MdlSpinnerModule,
        MdlSwitchModule,
        MdlTableModule,
        MdlTabsModule,
        MdlTextFieldModule,
        MdlTooltipModule], exports: [MdlBadgeModule,
        MdlButtonModule,
        MdlCardModule,
        MdlCheckboxModule,
        MdlChipModule,
        MdlCommonsModule,
        MdlDialogModule,
        MdlDialogOutletModule,
        MdlIconModule,
        MdlIconToggleModule,
        MdlLayoutModule,
        MdlListModule,
        MdlMenuModule,
        MdlProgressModule,
        MdlRadioModule,
        MdlRippleModule,
        MdlShadowModule,
        MdlSliderModule,
        MdlSnackbarModule,
        MdlSpinnerModule,
        MdlSwitchModule,
        MdlTableModule,
        MdlTabsModule,
        MdlTextFieldModule,
        MdlTooltipModule] });
MdlNonRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlNonRootModule, imports: [MODULES, MdlBadgeModule,
        MdlButtonModule,
        MdlCardModule,
        MdlCheckboxModule,
        MdlChipModule,
        MdlCommonsModule,
        MdlDialogModule,
        MdlDialogOutletModule,
        MdlIconModule,
        MdlIconToggleModule,
        MdlLayoutModule,
        MdlListModule,
        MdlMenuModule,
        MdlProgressModule,
        MdlRadioModule,
        MdlRippleModule,
        MdlShadowModule,
        MdlSliderModule,
        MdlSnackbarModule,
        MdlSpinnerModule,
        MdlSwitchModule,
        MdlTableModule,
        MdlTabsModule,
        MdlTextFieldModule,
        MdlTooltipModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlNonRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: MODULES,
                    exports: MODULES,
                }]
        }] });
