// @angular-mdl/core/esm2020/lib/snackbar/mdl-snackbar.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { MdlDialogOutletModule } from "../dialog-outlet/mdl-dialog-outlet.module";
import { MdlSnackbarComponent, MdlSnackbarService, } from "./mdl-snackbar.service";
import * as i0 from "@angular/core";
import * as i1 from "../dialog-outlet/mdl-dialog-outlet.module";
export class MdlSnackbarModule {
    static forRoot() {
        return {
            ngModule: MdlSnackbarModule,
            providers: [MdlSnackbarService],
        };
    }
}
MdlSnackbarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSnackbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlSnackbarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlSnackbarModule, declarations: [MdlSnackbarComponent], imports: [CommonModule, i1.MdlDialogOutletModule], exports: [MdlSnackbarComponent] });
MdlSnackbarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSnackbarModule, imports: [CommonModule, MdlDialogOutletModule.forRoot()] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSnackbarModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, MdlDialogOutletModule.forRoot()],
                    exports: [MdlSnackbarComponent],
                    declarations: [MdlSnackbarComponent],
                }]
        }] });
export * from "./mdl-snackbar.service";
