// @angular-mdl/core/esm2020/lib/spinner/mdl-spinner.component.mjs
import { Component, HostBinding, Input, ViewEncapsulation, } from "@angular/core";
import { toBoolean } from "../common/boolean-property";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MdlSpinnerComponent {
    constructor() {
        this.isSpinner = true;
        this.isUpgraded = true;
        this.layers = [1, 2, 3, 4];
        this.activeIntern = false;
        this.singleColorIntern = false;
    }
    get active() {
        return this.activeIntern;
    }
    set active(value) {
        this.activeIntern = toBoolean(value);
    }
    get singleColor() {
        return this.singleColorIntern;
    }
    set singleColor(value) {
        this.singleColorIntern = toBoolean(value);
    }
}
MdlSpinnerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSpinnerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlSpinnerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlSpinnerComponent, selector: "mdl-spinner", inputs: { active: "active", singleColor: ["single-color", "singleColor"] }, host: { properties: { "class.mdl-spinner": "this.isSpinner", "class.is-upgraded": "this.isUpgraded", "class.is-active": "this.active", "class.mdl-spinner--single-color": "this.singleColor" } }, ngImport: i0, template: `
    <div
      *ngFor="let layer of layers"
      class="mdl-spinner__layer mdl-spinner__layer-{{ layer }}"
    >
      <div class="mdl-spinner__circle-clipper mdl-spinner__left">
        <div class="mdl-spinner__circle"></div>
      </div>
      <div class="mdl-spinner__gap-patch">
        <div class="mdl-spinner__circle"></div>
      </div>
      <div class="mdl-spinner__circle-clipper mdl-spinner__right">
        <div class="mdl-spinner__circle"></div>
      </div>
    </div>
  `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSpinnerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-spinner",
                    template: `
    <div
      *ngFor="let layer of layers"
      class="mdl-spinner__layer mdl-spinner__layer-{{ layer }}"
    >
      <div class="mdl-spinner__circle-clipper mdl-spinner__left">
        <div class="mdl-spinner__circle"></div>
      </div>
      <div class="mdl-spinner__gap-patch">
        <div class="mdl-spinner__circle"></div>
      </div>
      <div class="mdl-spinner__circle-clipper mdl-spinner__right">
        <div class="mdl-spinner__circle"></div>
      </div>
    </div>
  `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isSpinner: [{
                type: HostBinding,
                args: ["class.mdl-spinner"]
            }], isUpgraded: [{
                type: HostBinding,
                args: ["class.is-upgraded"]
            }], active: [{
                type: HostBinding,
                args: ["class.is-active"]
            }, {
                type: Input
            }], singleColor: [{
                type: HostBinding,
                args: ["class.mdl-spinner--single-color"]
            }, {
                type: Input,
                args: ["single-color"]
            }] } });
