// @angular-mdl/core/esm2020/lib/spinner/mdl-spinner.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { MdlSpinnerComponent } from "./mdl-spinner.component";
import * as i0 from "@angular/core";
const MDL_SPINNER_DIRECTIVES = [MdlSpinnerComponent];
export class MdlSpinnerModule {
    static forRoot() {
        return {
            ngModule: MdlSpinnerModule,
            providers: [],
        };
    }
}
MdlSpinnerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSpinnerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlSpinnerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlSpinnerModule, declarations: [MdlSpinnerComponent], imports: [CommonModule], exports: [MdlSpinnerComponent] });
MdlSpinnerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSpinnerModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSpinnerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    exports: MDL_SPINNER_DIRECTIVES,
                    declarations: MDL_SPINNER_DIRECTIVES,
                }]
        }] });
export * from "./mdl-spinner.component";
