// @angular-mdl/core/esm2020/lib/switch/mdl-switch.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { FormsModule } from "@angular/forms";
import { MdlSwitchComponent } from "./mdl-switch.component";
import * as i0 from "@angular/core";
const MDL_SWITCH_DIRECTIVES = [MdlSwitchComponent];
export class MdlSwitchModule {
    static forRoot() {
        return {
            ngModule: MdlSwitchModule,
            providers: [],
        };
    }
}
MdlSwitchModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSwitchModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlSwitchModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlSwitchModule, declarations: [MdlSwitchComponent], imports: [CommonModule, FormsModule], exports: [MdlSwitchComponent] });
MdlSwitchModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSwitchModule, imports: [CommonModule, FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSwitchModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule],
                    exports: MDL_SWITCH_DIRECTIVES,
                    declarations: MDL_SWITCH_DIRECTIVES,
                }]
        }] });
export * from "./mdl-switch.component";
