// @angular-mdl/core/esm2020/lib/shadow/mdl-shadow.module.mjs
import { NgModule } from "@angular/core";
import { MdlShadowDirective } from "./mdl-shadow.directive";
import * as i0 from "@angular/core";
const MDL_SHADOW_DIRECTIVES = [MdlShadowDirective];
export class MdlShadowModule {
    static forRoot() {
        return {
            ngModule: MdlShadowModule,
            providers: [],
        };
    }
}
MdlShadowModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlShadowModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlShadowModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlShadowModule, declarations: [MdlShadowDirective], exports: [MdlShadowDirective] });
MdlShadowModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlShadowModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlShadowModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_SHADOW_DIRECTIVES,
                    declarations: MDL_SHADOW_DIRECTIVES,
                }]
        }] });
export * from "./mdl-shadow.directive";
