// @angular-mdl/core/esm2020/lib/list/mdl-list.module.mjs
import { NgModule } from "@angular/core";
import { MdlListComponent, MdlListItemAvatarDirective, MdlListItemComponent, MdlListItemIconDirective, MdlListItemPrimaryContentComponent, MdlListItemSecondaryActionComponent, MdlListItemSecondaryContentComponent, MdlListItemSecondaryInfoComponent, MdlListItemSubTitleComponent, MdlListItemTextBodyComponent, } from "./mdl-list.component";
import * as i0 from "@angular/core";
const MDL_LIST_DIRECTIVES = [
    MdlListComponent,
    MdlListItemComponent,
    MdlListItemPrimaryContentComponent,
    MdlListItemIconDirective,
    MdlListItemAvatarDirective,
    MdlListItemSecondaryContentComponent,
    MdlListItemSecondaryActionComponent,
    MdlListItemSubTitleComponent,
    MdlListItemSecondaryInfoComponent,
    MdlListItemTextBodyComponent,
];
export class MdlListModule {
    static forRoot() {
        return {
            ngModule: MdlListModule,
            providers: [],
        };
    }
}
MdlListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlListModule, declarations: [MdlListComponent,
        MdlListItemComponent,
        MdlListItemPrimaryContentComponent,
        MdlListItemIconDirective,
        MdlListItemAvatarDirective,
        MdlListItemSecondaryContentComponent,
        MdlListItemSecondaryActionComponent,
        MdlListItemSubTitleComponent,
        MdlListItemSecondaryInfoComponent,
        MdlListItemTextBodyComponent], exports: [MdlListComponent,
        MdlListItemComponent,
        MdlListItemPrimaryContentComponent,
        MdlListItemIconDirective,
        MdlListItemAvatarDirective,
        MdlListItemSecondaryContentComponent,
        MdlListItemSecondaryActionComponent,
        MdlListItemSubTitleComponent,
        MdlListItemSecondaryInfoComponent,
        MdlListItemTextBodyComponent] });
MdlListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_LIST_DIRECTIVES,
                    declarations: MDL_LIST_DIRECTIVES,
                }]
        }] });
export * from "./mdl-list.component";
