// @angular-mdl/core/esm2020/lib/list/mdl-list.component.mjs
import { Component, Directive, HostBinding, Input, Optional, ViewEncapsulation, } from "@angular/core";
import { MdlError, MdlStructureError } from "../common/mdl-error";
import { toNumber } from "../common/number.property";
import * as i0 from "@angular/core";
export class MdlUnsupportedCountOfListItemLinesError extends MdlError {
    constructor(lines) {
        /* istanbul ignore next */
        super(`"${lines}" is not supported - max 3 lines please.`);
    }
}
export class MdlListComponent {
    constructor() {
        this.isMdlList = true;
    }
}
MdlListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListComponent, selector: "mdl-list", host: { properties: { "class.mdl-list": "this.isMdlList" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isMdlList: [{
                type: HostBinding,
                args: ["class.mdl-list"]
            }] } });
export class MdlListItemComponent {
    constructor() {
        this.isMdlListItem = true;
        this.linesIntern = 1;
    }
    get lines() {
        return this.linesIntern;
    }
    set lines(value) {
        this.linesIntern = toNumber(value) ?? 1;
    }
    get lines2() {
        return this.lines === 2;
    }
    get lines3() {
        return this.lines === 3;
    }
    ngOnChanges() {
        if (this.lines && this.lines > 3) {
            throw new MdlUnsupportedCountOfListItemLinesError(this.lines);
        }
    }
}
MdlListItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlListItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemComponent, selector: "mdl-list-item", inputs: { lines: "lines" }, host: { properties: { "class.mdl-list__item": "this.isMdlListItem", "class.mdl-list__item--two-line": "this.lines2", "class.mdl-list__item--three-line": "this.lines3" } }, usesOnChanges: true, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { isMdlListItem: [{
                type: HostBinding,
                args: ["class.mdl-list__item"]
            }], lines: [{
                type: Input
            }], lines2: [{
                type: HostBinding,
                args: ["class.mdl-list__item--two-line"]
            }], lines3: [{
                type: HostBinding,
                args: ["class.mdl-list__item--three-line"]
            }] } });
export class MdlListItemPrimaryContentComponent {
    constructor(mdlListItemComponent) {
        this.mdlListItemComponent = mdlListItemComponent;
        this.isPrimaryContent = true;
    }
    ngOnInit() {
        if (this.mdlListItemComponent === null) {
            throw new MdlStructureError("mdl-list-item-primary-content", "mdl-list-item");
        }
    }
}
MdlListItemPrimaryContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemPrimaryContentComponent, deps: [{ token: MdlListItemComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlListItemPrimaryContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemPrimaryContentComponent, selector: "mdl-list-item-primary-content", host: { properties: { "class.mdl-list__item-primary-content": "this.isPrimaryContent" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemPrimaryContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item-primary-content",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlListItemComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isPrimaryContent: [{
                type: HostBinding,
                args: ["class.mdl-list__item-primary-content"]
            }] } });
export class MdlListItemSecondaryContentComponent {
    constructor(mdlListItemComponent) {
        this.mdlListItemComponent = mdlListItemComponent;
        this.isSecondaryContent = true;
    }
    ngOnInit() {
        if (this.mdlListItemComponent === null) {
            throw new MdlStructureError("mdl-list-item-secondary-content", "mdl-list-item");
        }
    }
}
MdlListItemSecondaryContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSecondaryContentComponent, deps: [{ token: MdlListItemComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlListItemSecondaryContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemSecondaryContentComponent, selector: "mdl-list-item-secondary-content", host: { properties: { "class.mdl-list__item-secondary-content": "this.isSecondaryContent" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSecondaryContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item-secondary-content",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlListItemComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isSecondaryContent: [{
                type: HostBinding,
                args: ["class.mdl-list__item-secondary-content"]
            }] } });
export class MdlListItemSecondaryActionComponent {
    constructor(mdlListItemComponent) {
        this.mdlListItemComponent = mdlListItemComponent;
        this.isSecondaryAction = true;
    }
    ngOnInit() {
        if (this.mdlListItemComponent === null) {
            throw new MdlStructureError("mdl-list-item-secondary-action", "mdl-list-item");
        }
    }
}
MdlListItemSecondaryActionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSecondaryActionComponent, deps: [{ token: MdlListItemComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlListItemSecondaryActionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemSecondaryActionComponent, selector: "mdl-list-item-secondary-action", host: { properties: { "class.mdl-list__item-secondary-action": "this.isSecondaryAction" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSecondaryActionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item-secondary-action",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlListItemComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isSecondaryAction: [{
                type: HostBinding,
                args: ["class.mdl-list__item-secondary-action"]
            }] } });
export class MdlListItemSubTitleComponent {
    constructor(mdlListItemComponent) {
        this.mdlListItemComponent = mdlListItemComponent;
        this.isSubTitle = true;
    }
    ngOnInit() {
        if (this.mdlListItemComponent === null) {
            throw new MdlStructureError("mdl-list-item-sub-title", "mdl-list-item-primary-content");
        }
    }
}
MdlListItemSubTitleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSubTitleComponent, deps: [{ token: MdlListItemPrimaryContentComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlListItemSubTitleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemSubTitleComponent, selector: "mdl-list-item-sub-title", host: { properties: { "class.mdl-list__item-sub-title": "this.isSubTitle" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSubTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item-sub-title",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlListItemPrimaryContentComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isSubTitle: [{
                type: HostBinding,
                args: ["class.mdl-list__item-sub-title"]
            }] } });
export class MdlListItemSecondaryInfoComponent {
    constructor(mdlListItemComponent) {
        this.mdlListItemComponent = mdlListItemComponent;
        this.isSecondaryInfo = true;
    }
    ngOnInit() {
        if (this.mdlListItemComponent === null) {
            throw new MdlStructureError("mdl-list-item-secondary-info", "mdl-list-item-secondary-content");
        }
    }
}
MdlListItemSecondaryInfoComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSecondaryInfoComponent, deps: [{ token: MdlListItemSecondaryContentComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlListItemSecondaryInfoComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemSecondaryInfoComponent, selector: "mdl-list-item-secondary-info", host: { properties: { "class.mdl-list__item-secondary-info": "this.isSecondaryInfo" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemSecondaryInfoComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item-secondary-info",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlListItemSecondaryContentComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isSecondaryInfo: [{
                type: HostBinding,
                args: ["class.mdl-list__item-secondary-info"]
            }] } });
export class MdlListItemTextBodyComponent {
    constructor(mdlListItemComponent) {
        this.mdlListItemComponent = mdlListItemComponent;
        this.isTextBody = true;
    }
    ngOnInit() {
        if (this.mdlListItemComponent === null) {
            throw new MdlStructureError("mdl-list-item-text-body", "mdl-list-item");
        }
    }
}
MdlListItemTextBodyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemTextBodyComponent, deps: [{ token: MdlListItemComponent, optional: true }], target: i0.ɵɵFactoryTarget.Component });
MdlListItemTextBodyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemTextBodyComponent, selector: "mdl-list-item-text-body", host: { properties: { "class.mdl-list__item-text-body": "this.isTextBody" } }, ngImport: i0, template: "<ng-content></ng-content>", isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemTextBodyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-list-item-text-body",
                    template: "<ng-content></ng-content>",
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: MdlListItemComponent, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { isTextBody: [{
                type: HostBinding,
                args: ["class.mdl-list__item-text-body"]
            }] } });
export class MdlListItemIconDirective {
    constructor() {
        this.isItemIcon = true;
    }
}
MdlListItemIconDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemIconDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlListItemIconDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemIconDirective, selector: "mdl-icon[mdl-list-item-icon]", host: { properties: { "class.mdl-list__item-icon": "this.isItemIcon" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemIconDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "mdl-icon[mdl-list-item-icon]",
                }]
        }], propDecorators: { isItemIcon: [{
                type: HostBinding,
                args: ["class.mdl-list__item-icon"]
            }] } });
export class MdlListItemAvatarDirective {
    constructor() {
        this.isItemAvatar = true;
    }
}
MdlListItemAvatarDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemAvatarDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdlListItemAvatarDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlListItemAvatarDirective, selector: "mdl-icon[mdl-list-item-avatar]", host: { properties: { "class.mdl-list__item-avatar": "this.isItemAvatar" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlListItemAvatarDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "mdl-icon[mdl-list-item-avatar]",
                }]
        }], propDecorators: { isItemAvatar: [{
                type: HostBinding,
                args: ["class.mdl-list__item-avatar"]
            }] } });
