// @angular-mdl/core/esm2020/lib/tooltip/mdl-tooltip-position.service.mjs
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
export class MdlTooltipPositionService {
    calcStyle(offsetWidth, offsetHeight, props, position) {
        const result = {};
        const left = props.left + props.width / 2;
        const top = props.top + props.height / 2;
        const marginLeft = -1 * (offsetWidth / 2);
        const marginTop = -1 * (offsetHeight / 2);
        if (position === "left" || position === "right") {
            if (top + marginTop < 0) {
                result["top"] = "0";
                result["marginTop"] = "0";
            }
            else {
                result["top"] = top + "px";
                result["marginTop"] = marginTop + "px";
            }
        }
        else {
            if (left + marginLeft < 0) {
                result["left"] = "0";
                result["marginLeft"] = "0";
            }
            else {
                result["left"] = left + "px";
                result["marginLeft"] = marginLeft + "px";
            }
        }
        if (position === "top") {
            result["top"] = props.top - offsetHeight - 10 + "px";
        }
        else if (position === "right") {
            result["left"] = props.left + props.width + 10 + "px";
        }
        else if (position === "left") {
            result["left"] = props.left - offsetWidth - 10 + "px";
        }
        else {
            result["top"] = props.top + props.height + 10 + "px";
        }
        return result;
    }
}
MdlTooltipPositionService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipPositionService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
MdlTooltipPositionService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipPositionService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipPositionService, decorators: [{
            type: Injectable
        }] });
