// @angular-mdl/core/esm2020/lib/tooltip/mdl-tooltip.module.mjs
import { NgModule } from "@angular/core";
import { MdlSimpleTooltipComponent, MdlTooltipComponent, } from "./mdl-tooltip.component";
import { MdlTooltipDirective, MdlTooltipLargeDirective, } from "./mdl-tooltip.directive";
import * as i0 from "@angular/core";
const MDL_TOOLTIP_DIRECTIVES = [
    MdlTooltipComponent,
    MdlTooltipLargeDirective,
    MdlTooltipDirective,
];
export * from "./mdl-tooltip.component";
export * from "./mdl-tooltip.directive";
export class MdlTooltipModule {
    static forRoot() {
        return {
            ngModule: MdlTooltipModule,
            providers: [],
        };
    }
}
MdlTooltipModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlTooltipModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipModule, declarations: [MdlTooltipComponent,
        MdlTooltipLargeDirective,
        MdlTooltipDirective, MdlSimpleTooltipComponent], exports: [MdlTooltipComponent,
        MdlTooltipLargeDirective,
        MdlTooltipDirective] });
MdlTooltipModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTooltipModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: MDL_TOOLTIP_DIRECTIVES,
                    declarations: [...MDL_TOOLTIP_DIRECTIVES, MdlSimpleTooltipComponent],
                }]
        }] });
