// @angular-mdl/core/esm2020/lib/common/append-view-container-ref-directive.mjs
import { Directive, Input, Renderer2, ViewContainerRef, } from "@angular/core";
import * as i0 from "@angular/core";
export class AppendViewContainerRefDirective {
    constructor(viewRef, renderer) {
        this.viewRef = viewRef;
        this.renderer = renderer;
    }
    ngAfterViewInit() {
        this.renderer.appendChild(this.viewRef.element.nativeElement, this.viewContainerRefToAppend?.element.nativeElement);
    }
}
AppendViewContainerRefDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AppendViewContainerRefDirective, deps: [{ token: i0.ViewContainerRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
AppendViewContainerRefDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: AppendViewContainerRefDirective, selector: "[append-view-container-ref]", inputs: { viewContainerRefToAppend: ["append-view-container-ref", "viewContainerRefToAppend"] }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AppendViewContainerRefDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: "[append-view-container-ref]",
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.Renderer2 }]; }, propDecorators: { viewContainerRefToAppend: [{
                type: Input,
                args: ["append-view-container-ref"]
            }] } });
