// @angular-mdl/core/esm2020/lib/common/mdl-common.module.mjs
import { NgModule } from "@angular/core";
import { AppendViewContainerRefDirective } from "./append-view-container-ref-directive";
import { Animations, NativeWebAnimations, NoopWebAnimations, } from "./animations";
import * as i0 from "@angular/core";
// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
function isWebAnimationsSupported() {
    return (typeof Element !== "undefined" &&
        typeof Element.prototype.animate === "function");
}
// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
export function instantiateSupportedAnimationDriver() {
    /* istanbul ignore next */
    if (isWebAnimationsSupported()) {
        return new NativeWebAnimations();
    }
    /* istanbul ignore next */
    return new NoopWebAnimations();
}
export class MdlCommonsModule {
}
MdlCommonsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCommonsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlCommonsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlCommonsModule, declarations: [AppendViewContainerRefDirective], exports: [AppendViewContainerRefDirective] });
MdlCommonsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCommonsModule, providers: [
        { provide: Animations, useFactory: instantiateSupportedAnimationDriver },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCommonsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: [AppendViewContainerRefDirective],
                    declarations: [AppendViewContainerRefDirective],
                    providers: [
                        { provide: Animations, useFactory: instantiateSupportedAnimationDriver },
                    ],
                }]
        }] });
export * from "./animations";
export * from "./boolean-property";
export * from "./mdl-error";
export * from "./append-view-container-ref-directive";
export * from "./native-support";
export * from "./noop";
export * from "./number.property";
