// @angular-mdl/core/esm2020/lib/chips/mdl-chip.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { MdlIconModule } from "../icon/mdl-icon.module";
import { MdlChipComponent } from "./mdl-chip.component";
import { MdlChipContactDirective } from "./mdl-chip-contact.directive";
import * as i0 from "@angular/core";
export * from "./mdl-chip.component";
export * from "./mdl-chip-contact.directive";
const DIRECTIVES = [MdlChipComponent, MdlChipContactDirective];
export class MdlChipModule {
    static forRoot() {
        return {
            ngModule: MdlChipModule,
            providers: [],
        };
    }
}
MdlChipModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlChipModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlChipModule, declarations: [MdlChipComponent, MdlChipContactDirective], imports: [MdlIconModule, CommonModule], exports: [MdlChipComponent, MdlChipContactDirective] });
MdlChipModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipModule, imports: [MdlIconModule, CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [MdlIconModule, CommonModule],
                    exports: DIRECTIVES,
                    declarations: DIRECTIVES,
                }]
        }] });
