// @angular-mdl/core/esm2020/lib/chips/mdl-chip-contact.directive.mjs
import { Directive, HostBinding, Inject, Optional, } from "@angular/core";
import { MdlChipComponent } from "./mdl-chip.component";
import { MdlStructureError } from "../common/mdl-error";
import * as i0 from "@angular/core";
import * as i1 from "./mdl-chip.component";
export class MdlChipContactDirective {
    constructor(mdlChipComponent) {
        this.mdlChipComponent = mdlChipComponent;
        this.isChipContact = true;
    }
    ngOnInit() {
        if (!this.mdlChipComponent) {
            throw new MdlStructureError("mdl-chip-contact", "mdl-chip");
        }
        this.mdlChipComponent.isChipContact = true;
    }
}
MdlChipContactDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipContactDirective, deps: [{ token: MdlChipComponent, optional: true }], target: i0.ɵɵFactoryTarget.Directive });
MdlChipContactDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: MdlChipContactDirective, selector: "[mdl-chip-contact]", host: { properties: { "class.mdl-chip__contact": "this.isChipContact" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipContactDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line
                    selector: '[mdl-chip-contact]'
                }]
        }], ctorParameters: function () { return [{ type: i1.MdlChipComponent, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [MdlChipComponent]
                }] }]; }, propDecorators: { isChipContact: [{
                type: HostBinding,
                args: ["class.mdl-chip__contact"]
            }] } });
