// @angular-mdl/core/esm2020/lib/chips/mdl-chip.component.mjs
import { Component, EventEmitter, HostBinding, Input, Output, ViewEncapsulation, } from "@angular/core";
import * as i0 from "@angular/core";
import * as i1 from "../icon/mdl-icon.component";
import * as i2 from "@angular/common";
export class MdlChipComponent {
    constructor() {
        // eslint-disable-next-line
        this.actionClick = new EventEmitter();
        this.isChip = true;
        this.isChipContact = false;
    }
    action() {
        this.actionClick.emit();
    }
}
MdlChipComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlChipComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlChipComponent, selector: "mdl-chip", inputs: { mdlLabel: ["mdl-label", "mdlLabel"], mdlActionIcon: ["mdl-action-icon", "mdlActionIcon"] }, outputs: { actionClick: "action-click" }, host: { properties: { "class.mdl-chip": "this.isChip", "class.mdl-chip--contact": "this.isChipContact" } }, ngImport: i0, template: `
    <ng-content></ng-content>
    <span *ngIf="mdlLabel" class="mdl-chip__text">{{ mdlLabel }}</span>
    <button
      *ngIf="mdlActionIcon"
      (click)="action()"
      type="button"
      class="mdl-chip__action"
    >
      <mdl-icon>{{ mdlActionIcon }}</mdl-icon>
    </button>
  `, isInline: true, dependencies: [{ kind: "component", type: i1.MdlIconComponent, selector: "mdl-icon" }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlChipComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-chip",
                    template: `
    <ng-content></ng-content>
    <span *ngIf="mdlLabel" class="mdl-chip__text">{{ mdlLabel }}</span>
    <button
      *ngIf="mdlActionIcon"
      (click)="action()"
      type="button"
      class="mdl-chip__action"
    >
      <mdl-icon>{{ mdlActionIcon }}</mdl-icon>
    </button>
  `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], propDecorators: { mdlLabel: [{
                type: Input,
                args: ["mdl-label"]
            }], mdlActionIcon: [{
                type: Input,
                args: ["mdl-action-icon"]
            }], actionClick: [{
                type: Output,
                args: ["action-click"]
            }], isChip: [{
                type: HostBinding,
                args: ["class.mdl-chip"]
            }], isChipContact: [{
                type: HostBinding,
                args: ["class.mdl-chip--contact"]
            }] } });
