// @angular-mdl/core/esm2020/lib/table/mdl-table.component.mjs
import { Component, EventEmitter, Input, Output, ViewEncapsulation, } from "@angular/core";
import * as i0 from "@angular/core";
import * as i1 from "../checkbox/mdl-checkbox.component";
import * as i2 from "@angular/common";
import * as i3 from "@angular/forms";
export class MdlDefaultTableModel {
    constructor(columns) {
        this.data = [];
        this.columns = columns;
    }
    addAll(data) {
        this.data.push(...data);
    }
}
const template = `
        <table class="mdl-data-table">
           <thead>
           <tr>
              <th *ngIf="selectable">
                 <mdl-checkbox mdl-ripple [ngModel]="isAllSelected()" (ngModelChange)="toggleAll()"></mdl-checkbox>
              </th>
              <th *ngFor="let column of model?.columns"
                  [ngClass]="{'mdl-data-table__cell--non-numeric': !column.numeric}">
                 {{column.name}}
              </th>
           </tr>
           </thead>
           <tbody>
           <tr *ngFor="let data of model?.data; let i = index" [ngClass]="{'is-selected': selectable && data.selected}">
              <td *ngIf="selectable">
                 <mdl-checkbox mdl-ripple
                      [(ngModel)]="data.selected"
                      (ngModelChange)="selectionChanged()"></mdl-checkbox>
              </td>
              <td *ngFor="let column of model?.columns"
                  [ngClass]="{'mdl-data-table__cell--non-numeric': !column.numeric}"
                  [innerHTML]="$any(data)[column.key]">
              </td>
           </tr>
           </tbody>
        </table>
    `;
const styles = `
    :host{
      display:inline-block;
    }
    `;
export class MdlTableComponent {
    constructor() {
        this.selectable = false;
    }
    isAllSelected() {
        return false;
    }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    toggleAll() { }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    selectionChanged() { }
}
MdlTableComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTableComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MdlTableComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlTableComponent, selector: "mdl-table", inputs: { model: ["table-model", "model"] }, ngImport: i0, template: "\n        <table class=\"mdl-data-table\">\n           <thead>\n           <tr>\n              <th *ngIf=\"selectable\">\n                 <mdl-checkbox mdl-ripple [ngModel]=\"isAllSelected()\" (ngModelChange)=\"toggleAll()\"></mdl-checkbox>\n              </th>\n              <th *ngFor=\"let column of model?.columns\"\n                  [ngClass]=\"{'mdl-data-table__cell--non-numeric': !column.numeric}\">\n                 {{column.name}}\n              </th>\n           </tr>\n           </thead>\n           <tbody>\n           <tr *ngFor=\"let data of model?.data; let i = index\" [ngClass]=\"{'is-selected': selectable && data.selected}\">\n              <td *ngIf=\"selectable\">\n                 <mdl-checkbox mdl-ripple\n                      [(ngModel)]=\"data.selected\"\n                      (ngModelChange)=\"selectionChanged()\"></mdl-checkbox>\n              </td>\n              <td *ngFor=\"let column of model?.columns\"\n                  [ngClass]=\"{'mdl-data-table__cell--non-numeric': !column.numeric}\"\n                  [innerHTML]=\"$any(data)[column.key]\">\n              </td>\n           </tr>\n           </tbody>\n        </table>\n    ", isInline: true, styles: [":host{display:inline-block}\n"], dependencies: [{ kind: "component", type: i1.MdlCheckboxComponent, selector: "mdl-checkbox", inputs: ["tabindex", "value", "disabled"], outputs: ["change"] }, { kind: "directive", type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i3.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i3.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlTableComponent, decorators: [{
            type: Component,
            args: [{ selector: "mdl-table", template: template, encapsulation: ViewEncapsulation.None, styles: [":host{display:inline-block}\n"] }]
        }], propDecorators: { model: [{
                type: Input,
                args: ["table-model"]
            }] } });
export class MdlSelectableTableComponent extends MdlTableComponent {
    constructor() {
        super(...arguments);
        // eslint-disable-next-line
        this.selected = [];
        // eslint-disable-next-line
        this.selectionChange = new EventEmitter();
        this.selectable = true;
        this.allSelected = false;
    }
    isAllSelected() {
        return this.model?.data.every((data) => data.selected) ?? false;
    }
    toggleAll() {
        const selected = !this.isAllSelected();
        this.model?.data.forEach((data) => (data.selected = selected));
        this.updateSelected();
    }
    selectionChanged() {
        this.updateSelected();
    }
    updateSelected() {
        this.selected = this.model?.data.filter((data) => data.selected) ?? [];
        this.selectionChange.emit({ value: this.selected });
    }
}
MdlSelectableTableComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSelectableTableComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
MdlSelectableTableComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlSelectableTableComponent, selector: "mdl-table-selectable", inputs: { model: ["table-model", "model"], selected: ["table-model-selected", "selected"] }, outputs: { selectionChange: "table-model-selectionChanged" }, usesInheritance: true, ngImport: i0, template: "\n        <table class=\"mdl-data-table\">\n           <thead>\n           <tr>\n              <th *ngIf=\"selectable\">\n                 <mdl-checkbox mdl-ripple [ngModel]=\"isAllSelected()\" (ngModelChange)=\"toggleAll()\"></mdl-checkbox>\n              </th>\n              <th *ngFor=\"let column of model?.columns\"\n                  [ngClass]=\"{'mdl-data-table__cell--non-numeric': !column.numeric}\">\n                 {{column.name}}\n              </th>\n           </tr>\n           </thead>\n           <tbody>\n           <tr *ngFor=\"let data of model?.data; let i = index\" [ngClass]=\"{'is-selected': selectable && data.selected}\">\n              <td *ngIf=\"selectable\">\n                 <mdl-checkbox mdl-ripple\n                      [(ngModel)]=\"data.selected\"\n                      (ngModelChange)=\"selectionChanged()\"></mdl-checkbox>\n              </td>\n              <td *ngFor=\"let column of model?.columns\"\n                  [ngClass]=\"{'mdl-data-table__cell--non-numeric': !column.numeric}\"\n                  [innerHTML]=\"$any(data)[column.key]\">\n              </td>\n           </tr>\n           </tbody>\n        </table>\n    ", isInline: true, styles: [":host{display:inline-block}\n"], dependencies: [{ kind: "component", type: i1.MdlCheckboxComponent, selector: "mdl-checkbox", inputs: ["tabindex", "value", "disabled"], outputs: ["change"] }, { kind: "directive", type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i3.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i3.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSelectableTableComponent, decorators: [{
            type: Component,
            args: [{ selector: "mdl-table-selectable", template: template, encapsulation: ViewEncapsulation.None, styles: [":host{display:inline-block}\n"] }]
        }], propDecorators: { model: [{
                type: Input,
                args: ["table-model"]
            }], selected: [{
                type: Input,
                args: ["table-model-selected"]
            }], selectionChange: [{
                type: Output,
                args: ["table-model-selectionChanged"]
            }] } });
