// @angular-mdl/core/esm2020/lib/checkbox/mdl-checkbox.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { FormsModule } from "@angular/forms";
import { MdlCheckboxComponent } from "./mdl-checkbox.component";
import * as i0 from "@angular/core";
const MDL_CHECKBOX_DIRECTIVES = [MdlCheckboxComponent];
export class MdlCheckboxModule {
    static forRoot() {
        return {
            ngModule: MdlCheckboxModule,
            providers: [],
        };
    }
}
MdlCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlCheckboxModule, declarations: [MdlCheckboxComponent], imports: [CommonModule, FormsModule], exports: [MdlCheckboxComponent] });
MdlCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCheckboxModule, imports: [CommonModule, FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule],
                    exports: MDL_CHECKBOX_DIRECTIVES,
                    declarations: MDL_CHECKBOX_DIRECTIVES,
                }]
        }] });
export * from "./mdl-checkbox.component";
