// @angular-mdl/core/esm2020/lib/dialog/internal-dialog-reference.mjs
import { Subject } from "rxjs";
import { MdlDialogReference } from "./mdl-dialog-reference";
/**
 * Internal representation of the dialog ref. the service
 * user should not have access to the created components
 * and internal implementations.
 */
export class InternalMdlDialogReference {
    constructor(config) {
        this.config = config;
        this.isModal = false;
        this.onHideSubject = new Subject();
        this.onVisibleSubject = new Subject();
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        this.closeCallback = () => { };
        this.dialogRef = new MdlDialogReference(this);
    }
    get hostDialog() {
        return this.hostDialogComponentRef?.instance;
    }
    hide(data) {
        this.onHideSubject.next(data);
        this.onHideSubject.complete();
        this.closeCallback();
    }
    visible() {
        this.onVisibleSubject.next();
        this.onVisibleSubject.complete();
    }
    onHide() {
        return this.onHideSubject.asObservable();
    }
    onVisible() {
        return this.onVisibleSubject.asObservable();
    }
}
