// @angular-mdl/core/esm2020/lib/dialog/mdl-dialog.component.mjs
import { Component, EventEmitter, Input, Output, TemplateRef, ViewChild, ViewEncapsulation, } from "@angular/core";
import { Subject } from "rxjs";
import { MdlDialogService } from "./mdl-dialog.service";
import * as i0 from "@angular/core";
import * as i1 from "./mdl-dialog.service";
export class MdlDialogComponent {
    constructor(dialogService) {
        this.dialogService = dialogService;
        // eslint-disable-next-line
        this.showEmitter = new EventEmitter();
        // eslint-disable-next-line
        this.hideEmitter = new EventEmitter();
        this.isShown = false;
        this.dialogRef = null;
    }
    show() {
        if (this.isShown) {
            throw new Error("Only one instance of an embedded mdl-dialog can exist!");
        }
        this.isShown = true;
        const mergedConfig = this.config || {};
        // default is true
        if (typeof mergedConfig.isModal === "undefined") {
            mergedConfig.isModal = true;
        }
        const result = new Subject();
        if (!this.template) {
            return result.asObservable();
        }
        const p = this.dialogService.showDialogTemplate(this.template, mergedConfig);
        p.subscribe((dialogRef) => {
            this.dialogRef = dialogRef;
            this.dialogRef.onVisible().subscribe(() => {
                this.showEmitter.emit(dialogRef);
                result.next(dialogRef);
                result.complete();
            });
            this.dialogRef.onHide().subscribe(() => {
                this.hideEmitter.emit();
                this.dialogRef = null;
                this.isShown = false;
            });
        });
        return result.asObservable();
    }
    close() {
        if (this.dialogRef) {
            this.dialogRef.hide();
        }
    }
}
MdlDialogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogComponent, deps: [{ token: i1.MdlDialogService }], target: i0.ɵɵFactoryTarget.Component });
MdlDialogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlDialogComponent, selector: "mdl-dialog", inputs: { config: ["mdl-dialog-config", "config"] }, outputs: { showEmitter: "show", hideEmitter: "hide" }, viewQueries: [{ propertyName: "template", first: true, predicate: TemplateRef, descendants: true, static: true }], ngImport: i0, template: `
    <div *dialogTemplate>
      <ng-content></ng-content>
    </div>
  `, isInline: true, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-dialog",
                    template: `
    <div *dialogTemplate>
      <ng-content></ng-content>
    </div>
  `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: i1.MdlDialogService }]; }, propDecorators: { template: [{
                type: ViewChild,
                args: [TemplateRef, { static: true }]
            }], config: [{
                type: Input,
                args: ["mdl-dialog-config"]
            }], showEmitter: [{
                type: Output,
                args: ["show"]
            }], hideEmitter: [{
                type: Output,
                args: ["hide"]
            }] } });
