// @angular-mdl/core/esm2020/lib/dialog/mdl-alert.component.mjs
import { Component, EventEmitter, HostBinding, Input, Output, } from "@angular/core";
import { MdlDialogService } from "./mdl-dialog.service";
import * as i0 from "@angular/core";
import * as i1 from "./mdl-dialog.service";
export class MdlAlertComponent {
    constructor(mdlDialogService) {
        this.mdlDialogService = mdlDialogService;
        this.message = "";
        this.display = "none";
        this.confirmed = new EventEmitter();
    }
    show() {
        this.mdlDialogService
            .alert(this.message, this.okText, this.title)
            .subscribe(() => {
            this.confirmed.emit();
        });
    }
}
MdlAlertComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlAlertComponent, deps: [{ token: i1.MdlDialogService }], target: i0.ɵɵFactoryTarget.Component });
MdlAlertComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlAlertComponent, selector: "mdl-alert", inputs: { title: "title", message: "message", okText: "okText" }, outputs: { confirmed: "confirmed" }, host: { properties: { "style.display": "this.display" } }, exportAs: ["mdlAlert"], ngImport: i0, template: ``, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlAlertComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-alert",
                    template: ``,
                    exportAs: "mdlAlert",
                }]
        }], ctorParameters: function () { return [{ type: i1.MdlDialogService }]; }, propDecorators: { title: [{
                type: Input
            }], message: [{
                type: Input
            }], okText: [{
                type: Input
            }], display: [{
                type: HostBinding,
                args: ["style.display"]
            }], confirmed: [{
                type: Output
            }] } });
