// @angular-mdl/core/esm2020/lib/dialog/mdl-simple-dialog.component.mjs
import { Component, forwardRef, HostListener, Inject, ViewChildren, ViewEncapsulation, } from "@angular/core";
import { MDL_CONFIGUARTION } from "./config";
import { MdlButtonComponent } from "../button/mdl-button.component";
import { MdlDialogReference } from "./mdl-dialog-reference";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../button/mdl-button.component";
import * as i3 from "./mdl-dialog-reference";
export class MdlSimpleDialogComponent {
    // why do i need forwardRef at this point, the demo LoginDialog dosn't need this!?!?
    constructor(dialogConfiguration, dialog) {
        this.dialogConfiguration = dialogConfiguration;
        this.dialog = dialog;
        this.dialog = dialog;
        dialog.onVisible().subscribe(() => {
            if (this.buttons) {
                this.buttons.first.elementRef.nativeElement.focus();
            }
        });
    }
    onEsc() {
        // run the first action that is marked as closing action
        const closeAction = this.dialogConfiguration.actions.find((action) => action.isClosingAction);
        if (closeAction) {
            closeAction.handler();
            this.dialog.hide();
        }
    }
    actionClicked(action) {
        action.handler();
        this.dialog.hide();
    }
}
MdlSimpleDialogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSimpleDialogComponent, deps: [{ token: forwardRef(() => MDL_CONFIGUARTION) }, { token: forwardRef(() => MdlDialogReference) }], target: i0.ɵɵFactoryTarget.Component });
MdlSimpleDialogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: MdlSimpleDialogComponent, selector: "mdl-dialog-component", host: { listeners: { "keydown.esc": "onEsc()" } }, viewQueries: [{ propertyName: "buttons", predicate: MdlButtonComponent, descendants: true }], ngImport: i0, template: `
    <h3 class="mdl-dialog__title" *ngIf="dialogConfiguration?.title">
      {{ dialogConfiguration?.title }}
    </h3>
    <div
      class="mdl-dialog__content"
      [innerHTML]="dialogConfiguration?.message"
    ></div>
    <div
      class="mdl-dialog__actions"
      [ngClass]="{
        'mdl-dialog__actions--full-width': dialogConfiguration?.fullWidthAction
      }"
    >
      <button
        mdl-button
        mdl-colored="primary"
        type="button"
        *ngFor="let action of dialogConfiguration?.actions"
        (click)="actionClicked(action)"
        [ngClass]="{ close: action.isClosingAction }"
      >
        {{ action.text }}
      </button>
    </div>
  `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i2.MdlButtonComponent, selector: "mdl-button, button[mdl-button], a[mdl-button]", inputs: ["mdl-button-type", "mdl-colored", "disabled"], exportAs: ["mdlButton"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlSimpleDialogComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "mdl-dialog-component",
                    template: `
    <h3 class="mdl-dialog__title" *ngIf="dialogConfiguration?.title">
      {{ dialogConfiguration?.title }}
    </h3>
    <div
      class="mdl-dialog__content"
      [innerHTML]="dialogConfiguration?.message"
    ></div>
    <div
      class="mdl-dialog__actions"
      [ngClass]="{
        'mdl-dialog__actions--full-width': dialogConfiguration?.fullWidthAction
      }"
    >
      <button
        mdl-button
        mdl-colored="primary"
        type="button"
        *ngFor="let action of dialogConfiguration?.actions"
        (click)="actionClicked(action)"
        [ngClass]="{ close: action.isClosingAction }"
      >
        {{ action.text }}
      </button>
    </div>
  `,
                    encapsulation: ViewEncapsulation.None,
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [forwardRef(() => MDL_CONFIGUARTION)]
                }] }, { type: i3.MdlDialogReference, decorators: [{
                    type: Inject,
                    args: [forwardRef(() => MdlDialogReference)]
                }] }]; }, propDecorators: { buttons: [{
                type: ViewChildren,
                args: [MdlButtonComponent]
            }], onEsc: [{
                type: HostListener,
                args: ["keydown.esc"]
            }] } });
