// @angular-mdl/core/esm2020/lib/dialog/mdl-dialog.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { MdlDialogService } from "./mdl-dialog.service";
import { MdlDialogComponent } from "./mdl-dialog.component";
import { MdlSimpleDialogComponent } from "./mdl-simple-dialog.component";
import { MdlDialogHostComponent } from "./mdl-dialog-host.component";
import { MdlAlertComponent } from "./mdl-alert.component";
import { MdlDialogOutletModule } from "../dialog-outlet/mdl-dialog-outlet.module";
import { MdlDialogOutletService } from "../dialog-outlet/mdl-dialog-outlet.service";
import { MdlCommonsModule } from "../common/mdl-common.module";
import { MdlButtonModule } from "../button/mdl-button.module";
import * as i0 from "@angular/core";
export * from "./mdl-dialog.component";
export * from "./mdl-dialog.service";
export * from "./mdl-dialog-configuration";
export * from "./mdl-alert.component";
export * from "./mdl-dialog-host.component";
export * from "./mdl-simple-dialog.component";
export * from "./mdl-dialog-reference";
const PUBLIC_COMPONENTS = [MdlDialogComponent, MdlAlertComponent];
const PRIVATE_COMPONENTS = [MdlDialogHostComponent, MdlSimpleDialogComponent];
export class MdlDialogModule {
    static forRoot() {
        return {
            ngModule: MdlDialogModule,
            providers: [MdlDialogService, MdlDialogOutletService],
        };
    }
}
MdlDialogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdlDialogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogModule, declarations: [MdlDialogComponent, MdlAlertComponent, MdlDialogHostComponent, MdlSimpleDialogComponent], imports: [CommonModule,
        MdlCommonsModule,
        MdlButtonModule,
        MdlDialogOutletModule], exports: [MdlDialogComponent, MdlAlertComponent] });
MdlDialogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogModule, imports: [CommonModule,
        MdlCommonsModule,
        MdlButtonModule,
        MdlDialogOutletModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: MdlDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        MdlCommonsModule,
                        MdlButtonModule,
                        MdlDialogOutletModule,
                    ],
                    exports: [...PUBLIC_COMPONENTS],
                    declarations: [...PUBLIC_COMPONENTS, ...PRIVATE_COMPONENTS],
                }]
        }] });
